<a id='beginning'></a>

# Карта ДТП

## Оглавление

1. [Введение](#intro)
2. [Описание данных](#description)
3. [Изучение и предработка данных](#data_pre)
4. [Исследовательский анализ данных](#data_analysis)
5. [Дополнительный анализ](#add_analysis)
6. [Итоговые выводы](#final_results)

<a id='intro'></a>

## Введение

### Описание исследования

Работа ведётся с данными «Карты ДТП» — некоммерческого проекта о проблеме дорожно-транспортных происшествий в России. Проект нацелен на повышение безопасности дорожного движения. «Карта ДТП» помогает определять действительные причины аварий, давать оценку уровню развития инфраструктуры, а также формировать качественные решения и программы по повышению безопасности улиц и дорог. А также подготовить рекомендации для заказчика по улучшению качества данных.



### Цель исследования

На основе анализа данных разработать меры по обеспечению безопасности дорожного движения.

### Задачи исследования

1. Ознакомиться с данными проекта «Карта ДТП»

2. Проверка ошибок в данных и их предобработка

3. Исследовательский анализ данных

    - Определить, сколько ДТП случалось в каждый день недели. Показать число происшествий по месяцам.

    - Выделить категории водителей по стажу. Определить, существуют ли категории, которые значительно отличаются по числу ДТП

<h2 id='description'>Описание данных</h2>

В этой тетрадке используются данные о ДТП в Москве и Красноярске.


#### 1. Основные датасеты о ДТП

**Файлы:**
- `Kirovskaya_oblast.csv`
- `Moscowskaya_oblast.csv`

Содержат информацию о дорожно-транспортных происшествиях.

| Поле | Описание |
|------|----------|
| `geometry.coordinates` | Координаты ДТП |
| `id` | Идентификатор ДТП |
| `properties.tags` | Тег происшествия |
| `properties.light` | Освещённость |
| `properties.point.lat` | Широта |
| `properties.point.long` | Долгота |
| `properties.nearby` | Ближайшие объекты |
| `properties.region` | Регион |
| `properties.scheme` | Схема ДТП |
| `properties.address` | Ближайший адрес |
| `properties.weather` | Погода |
| `properties.category` | Категория ДТП |
| `properties.datetime` | Дата и время ДТП |
| `properties.injured_count` | Число пострадавших |
| `properties.parent_region` | Область |
| `properties.road_conditions` | Состояние покрытия |
| `properties.participants_count` | Число участников |
| `properties.participant_categories` | Категории участников |

---

#### 2. Данные об участниках ДТП

**Файлы:**
- `Moscowskaya_oblast_participiants.csv`
- `Kirovskaya_oblast_participiants.csv`

Хранят сведения об участниках дорожно-транспортных происшествий.

| Поле | Описание |
|------|----------|
| `role` | Роль |
| `gender` | Пол |
| `violations` | Какие правила дорожного движения были нарушены конкретным участником |
| `health_status` | Состояние здоровья после ДТП |
| `years_of_driving_experience` | Число лет опыта |
| `id` | Идентификатор ДТП |

---

#### 3. Данные о транспортных средствах

**Файлы:**
- `Kirovskaya_oblast_vehicles.csv`
- `Moscowskaya_oblast_vehicles.csv`

Хранят сведения о транспортных средствах, участвовавших в ДТП.

| Поле | Описание |
|------|----------|
| `year` | Год выпуска |
| `brand` | Марка транспортного средства |
| `color` | Цвет |
| `model` | Модель |
| `category` | Категория |
| `id` | Идентификатор ДТП |

---


[Вернуть в начало подраздела 'Описание данных'](#description)

[Вернуться в начало тетрадки](#beginning)


<a id='data_pre'></a>

## Изучение и предработка данных

Изучение и предработка данных будет производиться на примере датасета `Kirovskaya_oblast.csv`

<h3 id='data'>Изучение данных</h3>

- [Загрузка библиотек](#load-libraries)
- [Загрузка данных](#load-data)
- [Обработка названий столбцов](#rename_columns)
- [Промежуточные выводы по данным](#conclus_data)
- [Предобработка данных](#pre)

<a id='load-libraries'></a>

Загружаем необходимые библиотеки для анализа данных

In [1]:
# Импортируем библиотеку pandas
import pandas as pd

# Импортируем библиотеку pathlib для работы с путями
from pathlib import Path

# Импортируем библиотеку datetime для работы с датами
import datetime

<a id='load-data'></a>

Загружаем данные из датасета `Kirovskaya_oblast.csv`

In [ ]:
# Выгружаем данные из датасета Kirovskaya_oblast.csv в датафрейм traf_accident_kirov

url_csv = '/**/datasets/Kirovskaya_oblast.csv'
local_csv = Path('datasets') / 'Kirovskaya_oblast.csv'

if local_csv.exists():
    # Если файл существует, то читаем его в датафрейм. Парсим дату и устанавливаем столбец id как индекс.
    traf_accident_kirov = pd.read_csv(
        local_csv, 
        parse_dates=['properties.datetime'], 
        index_col='id')
else:
    local_csv.parent.mkdir(parents=True, exist_ok=True)
    traf_accident_kirov = pd.read_csv(
        url_csv, 
        parse_dates=['properties.datetime'], 
        index_col='id')
    traf_accident_kirov.to_csv(local_csv, index=False)

In [3]:
# Выводим информацию о датафрейме
traf_accident_kirov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14517 entries, 1983180 to 1987515
Data columns (total 17 columns):
 #   Column                             Non-Null Count  Dtype         
---  ------                             --------------  -----         
 0   geometry.coordinates               14517 non-null  object        
 1   properties.tags                    14517 non-null  object        
 2   properties.light                   14517 non-null  object        
 3   properties.point.lat               14485 non-null  float64       
 4   properties.point.long              14485 non-null  float64       
 5   properties.nearby                  14517 non-null  object        
 6   properties.region                  14517 non-null  object        
 7   properties.scheme                  13380 non-null  float64       
 8   properties.address                 13843 non-null  object        
 9   properties.weather                 14517 non-null  object        
 10  properties.category            

<h3 id='rename_columns'>Обработаем названия столбцов</h3>

In [4]:
# Выводим название столбцов
list(traf_accident_kirov.columns)

['geometry.coordinates',
 'properties.tags',
 'properties.light',
 'properties.point.lat',
 'properties.point.long',
 'properties.nearby',
 'properties.region',
 'properties.scheme',
 'properties.address',
 'properties.weather',
 'properties.category',
 'properties.datetime',
 'properties.injured_count',
 'properties.parent_region',
 'properties.road_conditions',
 'properties.participants_count',
 'properties.participant_categories']

Удалим 'properties',  и приведем к 'змеиному' регистру, где это необходимо

In [5]:
# Обработаем названия столбцов
traf_accident_kirov.columns = traf_accident_kirov.columns.str.replace('properties.', '').str.replace('.', '_')

In [6]:
# Выводим название столбцов
list(traf_accident_kirov.columns)

['geometry_coordinates',
 'tags',
 'light',
 'point_lat',
 'point_long',
 'nearby',
 'region',
 'scheme',
 'address',
 'weather',
 'category',
 'datetime',
 'injured_count',
 'parent_region',
 'road_conditions',
 'participants_count',
 'participant_categories']

In [7]:
# Выводим первые 5 строк датафрейма
traf_accident_kirov.head()

,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,category,datetime,injured_count,parent_region,road_conditions,participants_count,participant_categories
id,,,,,,,,,,,,,,,,,
1983180,"[47.875603, 57.24379]",Дорожно-транспортные происшествия,Светлое время суток,57.243790,47.875603,[],Яранский район,600.0,Р-176 Вятка Чебоксары - Йошкар-Ола - Киров - С...,['Дождь'],Опрокидывание,2017-07-01 18:00:00,1,Кировская область,['Мокрое'],3,['Все участники']
2889433,"[47.87903, 57.304807]",Дорожно-транспортные происшествия,Светлое время суток,57.304807,47.879030,"['Административные здания', 'Нерегулируемый пе...",Яранский район,710.0,"г Яранск, ул Кирова, 10",['Ясно'],Наезд на пешехода,2023-09-12 17:10:00,1,Кировская область,"['Сухое', 'Отсутствие, плохая различимость гор...",2,"['Все участники', 'Пешеходы']"
2591208,"[47.840781, 57.297156]",Дорожно-транспортные происшествия,Сумерки,57.297156,47.840781,"['Жилые дома индивидуальной застройки', 'Нерег...",Яранский район,NaN,"г Яранск, ул Чапаева, 80",['Пасмурно'],Съезд с дороги,2021-07-02 21:25:00,1,Кировская область,['Мокрое'],1,['Все участники']
2577639,"[47.834365, 57.244775]",Дорожно-транспортные происшествия,Светлое время суток,57.244775,47.834365,['Жилые дома индивидуальной застройки'],Яранский район,200.0,"м Знаменка, ул Кирова, 15",['Пасмурно'],Столкновение,2021-05-31 18:55:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Мотоциклисты']"
1981026,"[47.968197, 57.357738]",Дорожно-транспортные происшествия,Светлое время суток,57.357738,47.968197,['Нерегулируемый перекрёсток неравнозначных ул...,Яранский район,NaN,"с/п Никольское, Киров-Советск- Яранск - подъез...",['Ясно'],Опрокидывание,2018-05-16 16:25:00,2,Кировская область,"['Сухое', 'Отсутствие, плохая различимость гор...",2,['Все участники']


In [8]:
# Выводим 5 последних строк датафрейма
traf_accident_kirov.tail()

,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,category,datetime,injured_count,parent_region,road_conditions,participants_count,participant_categories
id,,,,,,,,,,,,,,,,,
1981076,"[51.090943, 56.233566]",Дорожно-транспортные происшествия,Светлое время суток,56.233566,51.090943,['Жилые дома индивидуальной застройки'],Вятско-Полянский район,NaN,"г Вятские Поляны, ул Набережная, 6",['Ясно'],Иной вид ДТП,2018-05-17 06:20:00,1,Кировская область,['Сухое'],1,"['Все участники', 'Мотоциклисты']"
1985930,"[50.8911, 56.1978]",Дорожно-транспортные происшествия,Сумерки,56.197800,50.891100,[],Вятско-Полянский район,330.0,NaN,['Ясно'],Столкновение,2019-05-30 18:35:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Дети']"
1985949,"[51.079817, 56.22997]",Дорожно-транспортные происшествия,Светлое время суток,56.229970,51.079817,"['Многоквартирные жилые дома', 'Выезд с прилег...",Вятско-Полянский район,NaN,"г Вятские Поляны, ул Урицкого, 20",['Ясно'],Столкновение,2019-05-13 09:20:00,1,Кировская область,"['Сухое', 'Отсутствие, плохая различимость гор...",2,['Все участники']
1986807,"[51.165556, 56.152222]",Дорожно-транспортные происшествия,Светлое время суток,56.152222,51.165556,"['Многоквартирные жилые дома', 'Жилые дома инд...",Вятско-Полянский район,500.0,"г Сосновка, ул Пролетарская, 13",['Ясно'],Столкновение,2015-08-12 17:05:00,3,Кировская область,"['Сухое', 'Отсутствие, плохая различимость гор...",3,['Все участники']
1987515,"[51.1217, 56.2153]",Дорожно-транспортные происшествия,Светлое время суток,56.215300,51.121700,['Выезд с прилегающей территории'],Вятско-Полянский район,130.0,"Вятские Поляны - Сосновка, 1 км",['Ясно'],Столкновение,2016-08-22 18:05:00,1,Кировская область,['Сухое'],2,['Все участники']


In [9]:
# Выводим 5 случайных строк датафрейма
traf_accident_kirov.sample(5)

,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,category,datetime,injured_count,parent_region,road_conditions,participants_count,participant_categories
id,,,,,,,,,,,,,,,,,
1984321,"[49.658051, 58.588481]",Дорожно-транспортные происшествия,"В темное время суток, освещение включено",58.588481,49.658051,['Спортивные и развлекательные объекты'],Киров,60.0,"г Киров, пр-кт Октябрьский, 147-а",['Пасмурно'],Столкновение,2015-12-24 20:40:00,1,Кировская область,['Обработанное противогололедными материалами'],2,['Все участники']
2623902,"[50.14173, 58.720231]",Дорожно-транспортные происшествия,Светлое время суток,58.720231,50.141730,"['Административные здания', 'Нерегулируемый пе...",Слободской район,130.0,"д Стулово, ул Трактовая, 40",['Пасмурно'],Столкновение,2021-09-24 14:50:00,1,Кировская область,['Сухое'],2,['Все участники']
2941444,"[49.668707, 58.593868]",Дорожно-транспортные происшествия,Светлое время суток,58.593868,49.668707,['Крупный торговый объект (являющийся объектом...,Киров,500.0,"г Киров, ул Карла Маркса, 113",['Пасмурно'],Столкновение,2023-11-29 12:25:00,1,Кировская область,['Обработанное противогололедными материалами'...,2,['Все участники']
1984444,"[51.1039, 56.3958]",Дорожно-транспортные происшествия,"В темное время суток, освещение включено",56.395800,51.103900,"['Административные здания', 'Остановка обществ...",Вятско-Полянский район,740.0,"г Вятские Поляны, ул Тойменка, 135 29б",['Пасмурно'],Наезд на пешехода,2017-11-17 17:20:00,1,Кировская область,['Отсутствие элементов обустройства остановочн...,2,"['Все участники', 'Пешеходы']"
1988085,"[49.677601, 58.583422]",Дорожно-транспортные происшествия,Светлое время суток,58.583422,49.677601,"['Многоквартирные жилые дома', 'Выезд с прилег...",Киров,420.0,"г Киров, ул Блюхера, 32",['Пасмурно'],Столкновение,2020-05-23 11:50:00,1,Кировская область,['Сухое'],2,['Все участники']


In [10]:
# Выводим основные стат показатели датафрейма
traf_accident_kirov.describe()

,point_lat,point_long,scheme,datetime,injured_count,participants_count
count,14485.000000,14485.000000,13380.000000,14517,14517.000000,14517.000000
mean,58.360330,49.682865,549.431241,2019-11-06 18:52:13.258937600,1.315699,2.355721
min,0.490633,0.592429,10.000000,2015-01-01 03:45:00,1.000000,1.000000
25%,58.430055,49.583488,300.000000,2017-07-14 23:00:00,1.000000,2.000000
50%,58.587315,49.654762,610.000000,2019-10-12 10:50:00,1.000000,2.000000
75%,58.615341,49.833856,820.000000,2022-03-26 15:30:00,1.000000,3.000000
max,68.395800,65.537500,980.000000,2024-08-31 18:43:00,30.000000,30.000000
std,1.037014,1.196323,293.917023,NaN,0.793685,1.004847


<a id='conclus_data'></a>

**Промежуточные выводы по подразделу «Изучение данных»**

В ходе выгрузки данных были выполнены следующие шаги:

- Данные из файла `Kirovskaya_oblast.csv` загружены в датафрейм `traf_accident_kirov`.
- На этапе считывания данных:
  - столбец `properties.datetime` сразу преобразован в тип `datetime64[ns]` с помощью параметра `parse_dates`;
  - столбец `id` установлен в качестве индекса датафрейма с помощью параметра `index_col`.
- Названия столбцов приведены к «змеиному» регистру, из них удалён префикс `properties.` — для удобства дальнейшей работы.

Датафрейм `traf_accident_kirov` содержит **14 517 записей** и **17 столбцов** (без учёта индекса `id`).

**Типы данных:**

- `datetime64[ns]` — 1 столбец (`datetime`);
- `float64` — 3 столбца (`point_lat`, `point_long`, `scheme`);
- `int64` — 2 столбца (`injured_count`, `participants_count`);
- `object` — 11 столбцов (текстовые и списочные признаки).

Часть столбцов можно привести к корректным типам: 
- `point_lat`, `point_long` - в `float32`, 
- `injured_count`, `scheme` и `participants_count` — в `int16` для экономии памяти. 
-  `tags`, `nearby`, `weather`, `road_conditions`, `participant_categories` оставить `object`.

**Потенциальные аномалии по описательным статистикам:**

- Координаты имеют подозрительные минимумы (`lat = 0.49`, `long = 0.59`) и максимумы (`lat = 68.40`, `long = 65.54`), что выходит за пределы Кировской области (примерный диапазон: широта 55.5–61.0, долгота 45.5–53.5). Скорее всего, это ошибки ввода или геокодирования.
- `injured_count` и `participants_count` имеют максимум 30, что может быть как реальными крупными ДТП (например, авария с автобусом), так и выбросами — требует дополнительной проверки.
- `scheme` содержит значения от 10 до 980 — неизвестно, что соответствуют пикетажу / километражу участка дороги. Требуется дополнительная проверка.
- Временной диапазон данных: с **2015-01-01** по **2024-08-31** — около 10 лет наблюдений.

**Действия по обработке, которые потребуются на следующем этапе:**

1. Обработка пропусков в данных.
2. Приведение числовых столбцов к более компактным типам для экономии памяти.
3. Проверка дубликатов в данных.

В данных встречаются пропуски в столбцах `point_lat`, `point_long`, `scheme`, `address`.

Индекс и тип `datetime` установлены сразу на этапе считывания — это упростит дальнейший анализ. Для перехода к исследовательскому анализу необходима предобработка: обработка пропусков, оптимизация типов, обработка дубликатов.



[Вернуть в начало подраздела 'Изучение данных'](#data)

[Вернуться в начало тетрадки](#beginning)

<a id='pre'></a>

## Предобработка данных

В этом разделе посмотрим на количество пропущенных значений по столбцам датафрейма.

- [Вывод пропусков](#missing_values)
- [Промежуточные выводы по пропускам](#conclus_missing_values)
- [Обработка пропусков в данных](#missing_values_processing)
- [Корректировка типа данных](#data_type_correction)
- [Явные и неявные дубликаты](#duplicates)
- [Общий промежуточный вывод по предобработке](#conclus_pre)

<h3 id='missing_values'>Вывод пропусков</h3>

In [11]:
# Выводим количество пропущенных строк в датафрейме
traf_accident_kirov.isna().sum()

geometry_coordinates         0
tags                         0
light                        0
point_lat                   32
point_long                  32
nearby                       0
region                       0
scheme                    1137
address                    674
weather                      0
category                     0
datetime                     0
injured_count                0
parent_region                0
road_conditions              0
participants_count           0
participant_categories       0
dtype: int64

In [12]:
# Подсчитываем процент строк с пропусками
traf_accident_kirov.isna().sum() / len(traf_accident_kirov) * 100

geometry_coordinates      0.000000
tags                      0.000000
light                     0.000000
point_lat                 0.220431
point_long                0.220431
nearby                    0.000000
region                    0.000000
scheme                    7.832197
address                   4.642833
weather                   0.000000
category                  0.000000
datetime                  0.000000
injured_count             0.000000
parent_region             0.000000
road_conditions           0.000000
participants_count        0.000000
participant_categories    0.000000
dtype: float64

<a id='conclus_missing_values'></a>

**Промежуточные выводы по подразделу «Наличие пропусков в данных»**

| Столбец | Количество пропусков | Доля от всех строк |
|---------|----------------------|---------------------|
| `point_lat` | 32 | 0.22% |
| `point_long` | 32 | 0.22% |
| `scheme` | 1 137 | 7.83% |
| `address` | 674 | 4.64% |

Остальные поля заполнены полностью.

- **`point_lat`, `point_long`** — 32 пропуска (0.22%). Вероятная причина: ошибки при геокодировании либо отсутствие координат у конкретного ДТП. Вероятно, координаты можно восстановить из geometry_coordinates, если они дублируются. Если нет — 32 строки можно удалить.
- **`scheme`** — 1 137 пропусков (7.83%). По-видимому, схема / пикетаж не фиксировались для части ДТП (особенно это характерно для происшествий во дворах и на малых улицах). Рекомендуется уточнить у заказчика: в каких случаях схема не заполняется и насколько важно наличие этого поля - это поможет сформулировать рекомендации по сбору данных.
- **`address`** — 674 пропуска (4.64%). Поскольку среди строк с пропуском address есть непустые значения в nearby, похоже, заполнение address не является строго обязательным.

Поскольку доля мала и нет влияния на результаты анализа, данные записи можно оставить как есть.

[Вернуть в начало подраздела 'Предобработка данных'](#pre)

[Вернуться в начало тетрадки](#beginning)

<h3 id='missing_values_processing'>Обработка пропусков в данных</h3>

Расмотрим причины возникновения пропусков в данных и способы их избежать в дальнейшем.

In [13]:
# Пропуски в столбце scheme. 5 первых строк
traf_accident_kirov['scheme'].value_counts().head()

scheme
610.0    1051
300.0    1016
200.0     917
70.0      870
600.0     863
Name: count, dtype: int64

In [14]:
# Пропуски в столбце scheme. 5 последних строк
traf_accident_kirov['scheme'].value_counts().tail()

scheme
620.0    6
980.0    4
340.0    2
310.0    2
970.0    1
Name: count, dtype: int64

В столбце `Scheme` содержатся числовые коды, обозначающие схему конкретной аварии. Заполнять его трудоёмко, к тому же для текущего исследования он не требуется.

Лучше запросить у заказчика пояснения: в каких случаях схема ДТП может отсутствовать и насколько критично наличие этого поля в данных. Полученные уточнения помогут грамотно сформулировать рекомендации по процессу сбора информации.

In [15]:
# Пропуски в столбце address. Выведим первые 5 строк.
traf_accident_kirov[traf_accident_kirov['address'].isna()].head()

,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,category,datetime,injured_count,parent_region,road_conditions,participants_count,participant_categories
id,,,,,,,,,,,,,,,,,
1981028,"[47.703667, 57.398549]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",57.398549,47.703667,[],Яранский район,820.0,NaN,['Пасмурно'],Наезд на пешехода,2018-05-01 22:10:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Пешеходы']"
2875566,"[47.969474, 57.105731]",Дорожно-транспортные происшествия,Светлое время суток,57.105731,47.969474,[],Яранский район,950.0,NaN,['Ясно'],Опрокидывание,2023-08-13 16:20:00,1,Кировская область,['Сухое'],1,['Все участники']
2875597,"[47.891099, 57.309193]",Дорожно-транспортные происшествия,Светлое время суток,57.309193,47.891099,"['Многоквартирные жилые дома', 'Внутридворовая...",Яранский район,410.0,NaN,['Ясно'],Столкновение,2023-08-01 12:35:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Мотоциклисты']"
2861063,"[47.906871, 57.020891]",Дорожно-транспортные происшествия,Светлое время суток,57.020891,47.906871,[],Яранский район,600.0,NaN,['Пасмурно'],Съезд с дороги,2023-07-22 03:50:00,3,Кировская область,['Мокрое'],3,['Все участники']
1984517,"[47.828239, 57.298419]",Дорожно-транспортные происшествия,Светлое время суток,57.298419,47.828239,"['АЗС', 'Автостоянка (отделенная от проезжей ч...",Яранский район,830.0,NaN,['Ясно'],Наезд на пешехода,2020-06-25 08:50:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Пешеходы']"


In [16]:
# Пропуски в столбце address. Выведим последние 5 строк.
traf_accident_kirov[traf_accident_kirov['address'].isna()].tail()

,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,category,datetime,injured_count,parent_region,road_conditions,participants_count,participant_categories
id,,,,,,,,,,,,,,,,,
1983961,"[51.079023, 56.227214]",Дорожно-транспортные происшествия,Светлое время суток,56.227214,51.079023,"['Многоквартирные жилые дома', 'Внутридворовая...",Вятско-Полянский район,880.0,NaN,['Ясно'],Наезд на пешехода,2020-04-18 10:30:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Пешеходы']"
1986098,"[51.293557, 56.251791]",Дорожно-транспортные происшествия,"В темное время суток, освещение не включено",56.251791,51.293557,"['Многоквартирные жилые дома', 'Внутридворовая...",Вятско-Полянский район,NaN,NaN,['Пасмурно'],Наезд на велосипедиста,2020-08-19 18:11:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Велосипедисты', 'Дети']"
1986546,"[51.072639, 56.221047]",Дорожно-транспортные происшествия,Светлое время суток,56.221047,51.072639,"['Многоквартирные жилые дома', 'Внутридворовая...",Вятско-Полянский район,880.0,NaN,['Ясно'],Наезд на пешехода,2019-09-03 11:20:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Пешеходы']"
1980230,"[51.1186, 56.2417]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",56.241700,51.118600,[],Вятско-Полянский район,NaN,NaN,['Ясно'],Наезд на препятствие,2018-07-02 19:25:00,1,Кировская область,"['Сухое', 'Дефекты покрытия']",1,"['Все участники', 'Мотоциклисты']"
1985930,"[50.8911, 56.1978]",Дорожно-транспортные происшествия,Сумерки,56.197800,50.891100,[],Вятско-Полянский район,330.0,NaN,['Ясно'],Столкновение,2019-05-30 18:35:00,1,Кировская область,['Сухое'],2,"['Все участники', 'Дети']"


В столбце `address` встречается 674 пропусков. Ввероятно, адрес просто не был указан.

Стоит задуматься, есть ли какая‑то специфика у ДТП, для которых нет адреса: например, могло ли это быть связано с отсутствием поблизости значимых объектов. Получить основания для подтверждения или опровержения такой гипотезы не представляется возможным. Похоже, заполнение столбца `address` не является строго обязательным.


Количество пропусков в столбцах с координатами ничтожно мало (22 в каждом). Вероятная причина: ошибки при геокодировании либо отсутствие координат у конкретного ДТП. Обработка данных не требуется.

[Вернуть в начало подраздела 'Предобработка данных'](#pre)

[Вернуться в начало тетрадки](#beginning)

<h3 id='data_type_correction'>Корректировка типа данных</h3>

Для оптимизации возможно применение числовых типов с уменьшенной разрядностью.

- `float32`: `point_lat`, `point_long`, `scheme`
- `int16`: `injured_count`, `participants_count`

In [17]:
# Преобразовываем типы данных с понижением разрядности
traf_accident_kirov = traf_accident_kirov.astype({
    'injured_count': 'int16',
    'participants_count': 'int16',
    'point_lat': 'float32',
    'point_long': 'float32',
    'scheme': 'float32'
})

In [18]:
# Выводим информацию по traf_accident_kirov
traf_accident_kirov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14517 entries, 1983180 to 1987515
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   geometry_coordinates    14517 non-null  object        
 1   tags                    14517 non-null  object        
 2   light                   14517 non-null  object        
 3   point_lat               14485 non-null  float32       
 4   point_long              14485 non-null  float32       
 5   nearby                  14517 non-null  object        
 6   region                  14517 non-null  object        
 7   scheme                  13380 non-null  float32       
 8   address                 13843 non-null  object        
 9   weather                 14517 non-null  object        
 10  category                14517 non-null  object        
 11  datetime                14517 non-null  datetime64[ns]
 12  injured_count           14517 non-null  int

In [19]:
# Выводим количество пропущенных строк в датафрейме
traf_accident_kirov.isna().sum()

geometry_coordinates         0
tags                         0
light                        0
point_lat                   32
point_long                  32
nearby                       0
region                       0
scheme                    1137
address                    674
weather                      0
category                     0
datetime                     0
injured_count                0
parent_region                0
road_conditions              0
participants_count           0
participant_categories       0
dtype: int64

In [20]:
# Подсчитываем процент строк с пропусками
traf_accident_kirov.isna().sum() / len(traf_accident_kirov) * 100

geometry_coordinates      0.000000
tags                      0.000000
light                     0.000000
point_lat                 0.220431
point_long                0.220431
nearby                    0.000000
region                    0.000000
scheme                    7.832197
address                   4.642833
weather                   0.000000
category                  0.000000
datetime                  0.000000
injured_count             0.000000
parent_region             0.000000
road_conditions           0.000000
participants_count        0.000000
participant_categories    0.000000
dtype: float64




[Вернуть в начало подраздела 'Предобработка данных'](#pre)

[Вернуться в начало тетрадки](#beginning)

<h3 id='duplicates'>Явные и неявные дубликаты</h3>

Исследуем данные на предмет явных дубликатов

In [21]:
# Сортируем датафрейм
traf_accident_kirov = traf_accident_kirov.sort_values(by=list(traf_accident_kirov.columns))

In [22]:
# Выводим количество дубликатов
traf_accident_kirov.duplicated().sum()

np.int64(0)

Явные дубликаты отсутствуют

Исследуем данные на предмет неявных дубликатоы

In [23]:
traf_accident_kirov[traf_accident_kirov.duplicated(subset=['point_lat', 'point_long', 'datetime'])]

,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,category,datetime,injured_count,parent_region,road_conditions,participants_count,participant_categories
id,,,,,,,,,,,,,,,,,


ДТП с совпадающими координатами и временем отсутствуют. А значит и не являются дубликатами.

[Вернуть в начало подраздела 'Предобработка данных'](#pre)

[Вернуться в начало тетрадки](#beginning)

In [24]:
# Выводим информации о датафрейме
traf_accident_kirov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14517 entries, 1982256 to 1985738
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   geometry_coordinates    14517 non-null  object        
 1   tags                    14517 non-null  object        
 2   light                   14517 non-null  object        
 3   point_lat               14485 non-null  float32       
 4   point_long              14485 non-null  float32       
 5   nearby                  14517 non-null  object        
 6   region                  14517 non-null  object        
 7   scheme                  13380 non-null  float32       
 8   address                 13843 non-null  object        
 9   weather                 14517 non-null  object        
 10  category                14517 non-null  object        
 11  datetime                14517 non-null  datetime64[ns]
 12  injured_count           14517 non-null  int

<h3 id='conclus_pre'>Общий промежуточный вывод по предобработке</h3>

В ходе предобработки датасета `Kirovskaya_oblast.csv` были выполнены следующие шаги:

1. **Загрузка данных:** датафрейм `traf_accident_kirov` загружен с парсингом `properties.datetime` в `datetime64[ns]` и установкой `id` в качестве индекса.
2. **Названия столбцов** приведены к «змеиному» регистру: удалён префикс `properties.`, точки заменены на подчёркивания.
3. **Пропуски проанализированы, но не заполнялись:**
   - `point_lat`, `point_long` — по 32 пропуска (0.22%), решено оставить без изменений;
   - `scheme` — 1 137 пропусков (7.83%), поле не требуется для текущего анализа;
   - `address` — 674 пропуска (4.64%), заполнение не является обязательным.
4. **Типы данных оптимизированы:**
   - `injured_count`, `participants_count` → `int16`;
   - `point_lat`, `point_long`, `scheme` → `float32`.
   Это снизило потребление памяти с 2.0+ MB до 1.7+ MB.
5. **Дубликаты проверены:**
   - явные дубликаты отсутствуют;
   - неявные дубликаты по (`point_lat`, `point_long`, `datetime`) не обнаружены.
6. **Итоговый объём:** 14 517 строк, 17 столбцов. Пропуски остались в `point_lat`, `point_long`, `scheme`, `address`.

Датафрейм готов к исследовательскому анализу.

[Вернуть в начало подраздела 'Предобработка данных'](#pre)

[Вернуться в начало тетрадки](#beginning)

<a id='data_analysis'></a>

## Исследовательский анализ данных

Определим, сколько ДТП случалось в каждый день недели. Также покажим число происшествий по месяцам.

In [25]:
# Добавляем столбцы с днем недели и месяцем
days_ru = {
    0: 'Пн', 1: 'Вт', 2: 'Ср', 3: 'Чт',
    4: 'Пт', 5: 'Сб', 6: 'Вс'
}

days_order = ['Пн', 'Вт', 'Ср', 'Чт', 'Пт', 'Сб', 'Вс']

traf_accident_kirov['day_of_week'] = pd.Categorical(
    traf_accident_kirov['datetime'].dt.dayofweek.map(days_ru),
    categories=days_order,
    ordered=True
)


months_ru = {
    1: 'Январь', 2: 'Февраль', 3: 'Март', 4: 'Апрель',
    5: 'Май', 6: 'Июнь', 7: 'Июль', 8: 'Август',
    9: 'Сентябрь', 10: 'Октябрь', 11: 'Ноябрь', 12: 'Декабрь'
}

months_order = [
    'Январь', 'Февраль', 'Март', 'Апрель',
    'Май', 'Июнь', 'Июль', 'Август',
    'Сентябрь', 'Октябрь', 'Ноябрь', 'Декабрь'
]

traf_accident_kirov['month'] = pd.Categorical(
    traf_accident_kirov['datetime'].dt.month.map(months_ru),
    categories=months_order,
    ordered=True
)

traf_accident_kirov.head()


,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,category,datetime,injured_count,parent_region,road_conditions,participants_count,participant_categories,day_of_week,month
id,,,,,,,,,,,,,,,,,,,
1982256,"[0.592429, 0.490633]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",0.490633,0.592429,['Нерегулируемый перекрёсток неравнозначных ул...,Мурашинский район,410.0,Р-176 Вятка Чебоксары - Йошкар-Ола - Киров - С...,['Ясно'],Столкновение,2015-09-25 22:40:00,1,Кировская область,['Сухое'],3,['Все участники'],Пт,Сентябрь
1987661,"[19.4203, 52.7378]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",52.737801,19.420300,['Многоквартирные жилые дома'],Белохолуницкий район,200.0,"г Белая Холуница, ул Пионерская, 1",['Снегопад'],Столкновение,2016-11-28 18:50:00,3,Кировская область,['Обработанное противогололедными материалами'],4,['Все участники'],Пн,Ноябрь
1981422,"[19.8267, 58.3142]",Дорожно-транспортные происшествия,Светлое время суток,58.314201,19.826700,"['Остановка общественного транспорта', 'Нерегу...",Кирово-Чепецки район,90.0,"Киров - Малмыж - Вятские Поляны, 36 км",['Ясно'],Столкновение,2017-08-19 18:10:00,1,Кировская область,['Сухое'],3,['Все участники'],Сб,Август
1987662,"[25.4044, 52.21]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",52.209999,25.404400,[],Белохолуницкий район,830.0,"Белая Холуница - Кирс, 21 км",['Пасмурно'],Наезд на пешехода,2016-11-22 19:50:00,1,Кировская область,"['Сухое', 'Отсутствие, плохая различимость гор...",2,"['Все участники', 'Пешеходы']",Вт,Ноябрь
1980380,"[28.2006, 42.4367]",Дорожно-транспортные происшествия,Светлое время суток,42.436699,28.200600,[],Белохолуницкий район,220.0,Киров – Б.Холуница – Омутнинск – Афанасьево – ...,['Снегопад'],Столкновение,2016-12-10 09:30:00,1,Кировская область,['Обработанное противогололедными материалами'],3,['Все участники'],Сб,Декабрь


In [26]:
# Расчитываем количество ДТП по дням недели
traf_accident_kirov.groupby('day_of_week', observed=True).size()


day_of_week
Пн    2010
Вт    1988
Ср    1939
Чт    1936
Пт    2344
Сб    2246
Вс    2054
dtype: int64

In [27]:
# Расчитываем количество ДТП по месяцам
traf_accident_kirov.groupby('month', observed=True).size()

month
Январь      1069
Февраль      808
Март         799
Апрель       875
Май         1189
Июнь        1421
Июль        1635
Август      1654
Сентябрь    1370
Октябрь     1333
Ноябрь      1200
Декабрь     1164
dtype: int64

In [28]:
traf_accident_kirov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14517 entries, 1982256 to 1985738
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   geometry_coordinates    14517 non-null  object        
 1   tags                    14517 non-null  object        
 2   light                   14517 non-null  object        
 3   point_lat               14485 non-null  float32       
 4   point_long              14485 non-null  float32       
 5   nearby                  14517 non-null  object        
 6   region                  14517 non-null  object        
 7   scheme                  13380 non-null  float32       
 8   address                 13843 non-null  object        
 9   weather                 14517 non-null  object        
 10  category                14517 non-null  object        
 11  datetime                14517 non-null  datetime64[ns]
 12  injured_count           14517 non-null  int

[Вернуться в начало тетрадки](#beginning)

Выделиим категории водителей по стажу. Определим, существуют ли категории, которые значительно отличаются по числу ДТП.

In [ ]:
# Выгружаем данные из датасета Kirovskaya_oblast_participiants.csv в датафрейм part_accident_kirov

url_csv = '/**/datasets/Kirovskaya_oblast_participiants.csv'
local_csv = Path('datasets') / 'Kirovskaya_oblast_participiants.csv'

if local_csv.exists():
    # Если файл существует, то читаем его в датафрейм. Устанавливаем столбец id как индекс.
    part_accident_kirov = pd.read_csv(
        local_csv,
        index_col='id')
else:
    local_csv.parent.mkdir(parents=True, exist_ok=True)
    part_accident_kirov = pd.read_csv(
        url_csv, 
        index_col='id')
    part_accident_kirov.to_csv(local_csv, index=False)

In [30]:
# Выводим информацию о датафрейме
part_accident_kirov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 31235 entries, 1983180 to 2847894
Data columns (total 5 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   role                         31235 non-null  object 
 1   gender                       30387 non-null  object 
 2   violations                   31235 non-null  object 
 3   health_status                31135 non-null  object 
 4   years_of_driving_experience  16909 non-null  float64
dtypes: float64(1), object(4)
memory usage: 1.4+ MB


In [31]:
# Выводим первые 5 строк датафрейма
part_accident_kirov.head()

,role,gender,violations,health_status,years_of_driving_experience
id,,,,,
1983180,Водитель,Мужской,['Несоответствие скорости конкретным условиям ...,"Раненый, находящийся (находившийся) на амбулат...",26.0
2889433,Водитель,Мужской,[],Не пострадал,34.0
2591208,Пассажир,Мужской,[],"Раненый, находящийся (находившийся) на амбула...",NaN
2591208,Пассажир,Мужской,[],"Раненый, находящийся (находившийся) на амбула...",NaN
2577639,Водитель,Мужской,[],Не пострадал,27.0


In [32]:
# Получаем срез только по водителю и сохраняем в новый датафрейм
part_accident_kirov_actual = part_accident_kirov[part_accident_kirov['role'] == 'Водитель'].copy()

In [33]:
# Выводим информацию о датафрейме
part_accident_kirov_actual.info()

<class 'pandas.core.frame.DataFrame'>
Index: 19753 entries, 1983180 to 2847894
Data columns (total 5 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   role                         19753 non-null  object 
 1   gender                       19018 non-null  object 
 2   violations                   19753 non-null  object 
 3   health_status                19661 non-null  object 
 4   years_of_driving_experience  16909 non-null  float64
dtypes: float64(1), object(4)
memory usage: 925.9+ KB


In [34]:
# Выводим первые 5 строк датафрейма
part_accident_kirov_actual.head()

,role,gender,violations,health_status,years_of_driving_experience
id,,,,,
1983180,Водитель,Мужской,['Несоответствие скорости конкретным условиям ...,"Раненый, находящийся (находившийся) на амбулат...",26.0
2889433,Водитель,Мужской,[],Не пострадал,34.0
2577639,Водитель,Мужской,[],Не пострадал,27.0
2930931,Водитель,Мужской,['Несоответствие скорости конкретным условиям ...,"Раненый, находящийся (находившийся) на стацион...",NaN
2930982,Водитель,Мужской,['Несоответствие скорости конкретным условиям ...,Не пострадал,5.0


In [35]:
# Выводим количество пропусков в датафрейме
part_accident_kirov_actual.isna().sum()

role                              0
gender                          735
violations                        0
health_status                    92
years_of_driving_experience    2844
dtype: int64

In [36]:
# Выводим долю пропусков в процентах
part_accident_kirov_actual.isna().sum() / len(part_accident_kirov_actual) * 100

role                            0.000000
gender                          3.720954
violations                      0.000000
health_status                   0.465752
years_of_driving_experience    14.397813
dtype: float64

In [37]:
# Категоризация данных стажу вождения
part_accident_kirov_actual['category_exp'] = pd.cut(
    part_accident_kirov_actual['years_of_driving_experience'],
    [0, 5, 10, 15, 20, 25, 30, 35, 40, 50, 60, 70]
)


In [38]:
part_accident_kirov_actual['category_exp'].value_counts()

category_exp
(0, 5]      3533
(5, 10]     3238
(10, 15]    2769
(15, 20]    2280
(20, 25]    1587
(25, 30]    1137
(30, 35]     921
(35, 40]     697
(40, 50]     629
(50, 60]     106
(60, 70]      12
Name: count, dtype: int64

Наблюдается тенденция: чем больше стаж вождения, тем меньше количество аварий. Однако для корректных выводов нужно знать, сколько водителей входит в каждую группу по стажу, а таких данных у нас нет.

Рекомендации заказчику: сделать столбец со стажем вождения обязательным для заполнения. На данный момент в нём 2844 пропуска — около 14% от общего объёма данных. Случаи, когда у участника ДТП не было водительских прав, стоит фиксировать отдельно. Кроме того, полезно отмечать причину отсутствия прав: изъяты ли они были или водитель их так и не получил — вероятнее всего, этот фактор тоже имеет значение для анализа.

[Вернуться в начало тетрадки](#beginning)

<a id='add_analysis'></a>

## Дополнительный анализ

<a id='populate'></a>

### Анализ численности населения

Проанализируем, как менялось число аварий по годам в Кировской и Московской областях. При этом включим город Киров в соответствующую область, а Москву исключите. Также определим, в какой области больше аварий в относительном выражении — в пересчёте на 100 тысяч жителей.

In [ ]:
# Выгружаем данные из датасета Moscowskaya_oblast.csv в датафрейм traf_accident_moscow

url_csv = '/**/datasets/Moscowskaya_oblast.csv'
local_csv = Path('datasets') / 'Moscowskaya_oblast.csv'

if local_csv.exists():
    # Если файл существует, то читаем его в датафрейм. Парсим дату и устанавливаем столбец id как индекс.
    traf_accident_moscow = pd.read_csv(
        local_csv, 
        parse_dates=['properties.datetime'], 
        index_col='id')
else:
    local_csv.parent.mkdir(parents=True, exist_ok=True)
    traf_accident_moscow = pd.read_csv(
        url_csv, 
        parse_dates=['properties.datetime'], 
        index_col='id')
    traf_accident_moscow.to_csv(local_csv, index=False)

In [40]:
# Обработаем названия столбцов
traf_accident_moscow.columns = traf_accident_moscow.columns.str.replace('properties.', '').str.replace('.', '_')

In [41]:
# Преобразовываем типы данных с понижением разрядности
traf_accident_moscow = traf_accident_moscow.astype({
    'injured_count': 'int16',
    'participants_count': 'int16',
    'point_lat': 'float32',
    'point_long': 'float32',
    'scheme': 'float32'
})

In [42]:
# Выводим уникальные значения столбца parent_region в traf_accident_moscow
traf_accident_moscow['parent_region'].unique()

array(['Московская область'], dtype=object)

In [43]:
# Выводим информацию о датафрейме traf_accident_moscow
traf_accident_moscow.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45618 entries, 2163589 to 2130572
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   geometry_coordinates    45618 non-null  object        
 1   tags                    45618 non-null  object        
 2   light                   45618 non-null  object        
 3   point_lat               45615 non-null  float32       
 4   point_long              45615 non-null  float32       
 5   nearby                  45618 non-null  object        
 6   region                  45618 non-null  object        
 7   scheme                  44235 non-null  float32       
 8   address                 43862 non-null  object        
 9   weather                 45618 non-null  object        
 10  category                45618 non-null  object        
 11  datetime                45618 non-null  datetime64[ns]
 12  injured_count           45618 non-null  int

In [44]:
# Выводим уникальные значения столбца parent_region в датафрейме traf_accident_kirov
traf_accident_kirov['parent_region'].unique()

array(['Кировская область'], dtype=object)

In [45]:
# Выводим уникальные значения столбца region в датафрейме traf_accident_kirov
traf_accident_kirov['region'].unique()

array(['Мурашинский район', 'Белохолуницкий район',
       'Кирово-Чепецки район', 'Подосиновский район', 'Шабалинский район',
       'Кикнурский район', 'Санчурский район', 'Оричевский район',
       'Верхошижемский район', 'Даровской район', 'Лузский район',
       'Котельничский район', 'Свечинский район', 'Яранский район',
       'Опаринскийрайон', 'Тужинский район', 'Арбажский район',
       'Орловский район', 'Пижанский район', 'Юрьянский район', 'Киров',
       'Советский район', 'Уржумский район', 'Нолинский район',
       'Куменский район', 'Лебяжский район', 'Слободской район',
       'Богородский район', 'Сунский район', 'Малмыжский район',
       'Нагорский район', 'Зуевский район', 'Немский район',
       'Вятско-Полянский район', 'Кильмезский район', 'Омутнинский район',
       'Унинский район', 'Фаленский район', 'Верхнекамский район',
       'Афанасьевский район'], dtype=object)

In [46]:
# Выводим информацию о датафрейме traf_accident_kirov
traf_accident_kirov.info()

<class 'pandas.core.frame.DataFrame'>
Index: 14517 entries, 1982256 to 1985738
Data columns (total 19 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   geometry_coordinates    14517 non-null  object        
 1   tags                    14517 non-null  object        
 2   light                   14517 non-null  object        
 3   point_lat               14485 non-null  float32       
 4   point_long              14485 non-null  float32       
 5   nearby                  14517 non-null  object        
 6   region                  14517 non-null  object        
 7   scheme                  13380 non-null  float32       
 8   address                 13843 non-null  object        
 9   weather                 14517 non-null  object        
 10  category                14517 non-null  object        
 11  datetime                14517 non-null  datetime64[ns]
 12  injured_count           14517 non-null  int

In [47]:
# Добавляем столбец с годом в датафреймы
traf_accident_kirov['year'] = traf_accident_kirov['datetime'].dt.year
# Преобразовываем типы данных с понижением разрядности
traf_accident_kirov = traf_accident_kirov.astype({
    'year': 'int16'
})

traf_accident_moscow['year'] = traf_accident_moscow['datetime'].dt.year
# Преобразовываем типы данных с понижением разрядности
traf_accident_moscow = traf_accident_moscow.astype({
    'year': 'int16'
})

In [48]:
# Расчитываем количество ДТП по годам в traf_accident_kirov
traf_accident_kirov.groupby('year', observed=True).size()


year
2015    1532
2016    1468
2017    1452
2018    1612
2019    1568
2020    1498
2021    1524
2022    1386
2023    1486
2024     991
dtype: int64

In [49]:
# Расчитываем количество ДТП по годам в traf_accident_moscow
traf_accident_moscow.groupby('year', observed=True).size()

year
2015    6643
2016    5794
2017    5747
2018    5102
2019    5111
2020    4216
2021    3898
2022    3353
2023    3372
2024    2382
dtype: int64

Ниже приведена таблица с численностью постоянного населения по двум регионам за 2015–2024 гг. Данные взяты из официальных публикаций Росстата и обобщены в Википедии.

#### Численность населения, тыс. человек

| Год | Кировская область (включая г. Киров) | Московская область (без г. Москвы) |
|-----|--------------------------------------|-------------------------------------|
| 2015 | 1304 | 7231 |
| 2016 | 1297 | 7319 |
| 2017 | 1292 | 7423 |
| 2018 | 1283 | 7503 |
| 2019 | 1272 | 7600 |
| 2020 | 1262 | 7691 |
| 2021 | 1154 | 8525 |
| 2022 | 1149 | 8542 |
| 2023 | 1138 | 8592 |
| 2024 | 1130 | 8651 |

**Примечания:**
- Для Кировской области данные приведены с учётом города Киров (областной центр входит в состав области).
- Для Московской области данные приведены без города Москвы (Москва — отдельный субъект РФ).
- Значения округлены до целых тысяч человек.


**Источники:** Росстат; сводные данные по регионам (Википедия: «Население Кировской области», «Население Московской области»).

In [50]:
years = [2015, 2016, 2017, 2018, 2019, 2020, 2021, 2022, 2023, 2024]

# Кировская область
population_kirov = [1304, 1297, 1292, 1283, 1272, 1262, 1154, 1149, 1138, 1130]

kirov_dtp_on_years = pd.DataFrame({'year': years})
kirov_dtp_on_years['count'] = (
    kirov_dtp_on_years['year']
    .map(traf_accident_kirov.groupby('year', observed=True).size())
)
kirov_dtp_on_years['population'] = population_kirov
kirov_dtp_on_years['dtp_on_100k'] = (
    kirov_dtp_on_years['count'] / kirov_dtp_on_years['population'] * 100
).round(2)

# Московская область
population_moscow = [7231, 7319, 7423, 7503, 7600, 7691, 8525, 8542, 8592, 8651]

moscow_dtp_on_years = pd.DataFrame({'year': years})
moscow_dtp_on_years['count'] = (
    moscow_dtp_on_years['year']
    .map(traf_accident_moscow.groupby('year', observed=True).size())
)
moscow_dtp_on_years['population'] = population_moscow
moscow_dtp_on_years['dtp_on_100k'] = (
    moscow_dtp_on_years['count'] / moscow_dtp_on_years['population'] * 100
).round(2)

In [51]:
# Выводим данные по Кировской области
kirov_dtp_on_years

,year,count,population,dtp_on_100k
0,2015,1532,1304,117.48
1,2016,1468,1297,113.18
2,2017,1452,1292,112.38
3,2018,1612,1283,125.64
4,2019,1568,1272,123.27
5,2020,1498,1262,118.70
6,2021,1524,1154,132.06
7,2022,1386,1149,120.63
8,2023,1486,1138,130.58
9,2024,991,1130,87.70


In [52]:
# Выводим данные по Московской области
moscow_dtp_on_years

,year,count,population,dtp_on_100k
0,2015,6643,7231,91.87
1,2016,5794,7319,79.16
2,2017,5747,7423,77.42
3,2018,5102,7503,68.00
4,2019,5111,7600,67.25
5,2020,4216,7691,54.82
6,2021,3898,8525,45.72
7,2022,3353,8542,39.25
8,2023,3372,8592,39.25
9,2024,2382,8651,27.53


[Вернуться в начало тетрадки](#beginning)

<a id='weat_cond'></a>

### Анализ категорий аварий и погодных условий (по Кировской области)

In [53]:
# Вывести уникальные значения категорий аварий
traf_accident_kirov['category'].unique()

array(['Столкновение', 'Наезд на пешехода', 'Опрокидывание',
       'Наезд на животное', 'Отбрасывание предмета', 'Съезд с дороги',
       'Иной вид ДТП', 'Наезд на препятствие', 'Наезд на стоящее ТС',
       'Наезд на велосипедиста', 'Падение пассажира',
       'Наезд на лицо, не являющееся участником дорожного движения, осуществляющее производство работ',
       'Наезд на лицо, не являющееся участником дорожного движения, осуществляющее несение службы',
       'Наезд на внезапно возникшее препятствие', 'Падение груза',
       'Наезд на лицо, не являющееся участником дорожного движения, осуществляющее какую-либо другую деятельность'],
      dtype=object)

In [54]:
# Выводим количество аварий по категориям аварий
traf_accident_kirov['category'].value_counts()

category
Столкновение                                                                                                 5423
Наезд на пешехода                                                                                            4364
Съезд с дороги                                                                                               1155
Опрокидывание                                                                                                1002
Падение пассажира                                                                                             738
Наезд на препятствие                                                                                          667
Наезд на велосипедиста                                                                                        589
Наезд на стоящее ТС                                                                                           312
Иной вид ДТП                                                                   

#### Результат группировки категорий аварий

| Группа | Состав | Количество |
|---|---|---|
| Столкновение | Столкновение | 5423 |
| Наезд на пешехода | Наезд на пешехода + Наезд на велосипедиста | 4953 |
| Съезд с дороги | Съезд с дороги | 1155 |
| Опрокидывание | Опрокидывание | 1002 |
| Наезд иной | Наезд на препятствие + Наезд на стоящее ТС | 979 |
| Падение пассажира | Падение пассажира | 738 |
| Редкие | все остальные категории | 267 |



In [55]:
# Пользовательская группировка

def remap_category(cat):
    if cat == 'Столкновение':
        return 'Столкновение'
    elif cat in ('Наезд на пешехода', 'Наезд на велосипедиста'):
        return 'Наезд на пешехода'          # велосипедисты объединены с пешеходами
    elif cat == 'Съезд с дороги':
        return 'Съезд с дороги'
    elif cat == 'Опрокидывание':
        return 'Опрокидывание'
    elif cat == 'Падение пассажира':
        return 'Падение пассажира'
    elif cat in ('Наезд на препятствие', 'Наезд на стоящее ТС'):
        return 'Наезд иной'                  # препятствие + стоящее ТС → «Наезд иной»
    else:
        return 'Редкие'                      # все остальные категории

grouped_cat_dtp = traf_accident_kirov['category'].value_counts().groupby(remap_category).sum().sort_values(ascending=False)
grouped_cat_dtp


category
Столкновение         5423
Наезд на пешехода    4953
Съезд с дороги       1155
Опрокидывание        1002
Наезд иной            979
Падение пассажира     738
Редкие                267
Name: count, dtype: int64

In [56]:
# Вывести уникальные значения погоды
traf_accident_kirov['weather'].unique()

array(["['Ясно']", "['Снегопад']", "['Пасмурно']", "['Дождь']",
       "['Пасмурно', 'Снегопад']", "['Метель']", "['Пасмурно', 'Метель']",
       "['Пасмурно', 'Температура ниже -30С']", "['Пасмурно', 'Дождь']",
       "['Ясно', 'Температура выше +30С']", "['Дождь', 'Снегопад']",
       "['Туман']", "['Пасмурно', 'Туман']",
       "['Ясно', 'Температура ниже -30С']", "['Снегопад', 'Метель']",
       "['Ясно', 'Туман']", "['Пасмурно', 'Ураганный ветер']",
       "['Дождь', 'Туман']"], dtype=object)

In [57]:
# Выводим количество аварий по погоде
traf_accident_kirov['weather'].value_counts()

weather
['Пасмурно']                             6701
['Ясно']                                 6264
['Снегопад']                              608
['Дождь']                                 593
['Пасмурно', 'Снегопад']                  108
['Пасмурно', 'Дождь']                      85
['Ясно', 'Температура выше +30С']          66
['Метель']                                 27
['Туман']                                  18
['Снегопад', 'Метель']                     14
['Ясно', 'Температура ниже -30С']          11
['Пасмурно', 'Температура ниже -30С']       6
['Пасмурно', 'Туман']                       5
['Дождь', 'Снегопад']                       4
['Пасмурно', 'Метель']                      3
['Пасмурно', 'Ураганный ветер']             2
['Ясно', 'Туман']                           1
['Дождь', 'Туман']                          1
Name: count, dtype: int64

### Что куда попало

| Группа | Комбинации |
|---|---|
| **Ясно** | `['Ясно']`, `['Ясно', 'Температура выше +30С']`, `['Ясно', 'Температура ниже -30С']` |
| **Пасмурно** | `['Пасмурно']`, `['Пасмурно', 'Температура ниже -30С']`, `['Пасмурно', 'Ураганный ветер']` |
| **Дождь** | `['Дождь']`, `['Пасмурно', 'Дождь']`, `['Дождь', 'Туман']` |
| **Снег/Метель** | `['Снегопад']`, `['Пасмурно', 'Снегопад']`, `['Метель']`, `['Пасмурно', 'Метель']`, `['Снегопад', 'Метель']` |
| **Смешанные осадки** | `['Дождь', 'Снегопад']` |
| **Туман** | `['Туман']`, `['Пасмурно', 'Туман']`, `['Ясно', 'Туман']` |

### Логика приоритетов

1. **Смешанные осадки** проверяются первыми — одновременное наличие дождя и снега/метели выделяется в отдельную группу, так как это самая опасная комбинация для вождения (гололёд + слякоть + плохая видимость).
2. **Дождь** — любая комбинация с дождём, но без снега. `['Дождь', 'Туман']` остаётся здесь, а не в «Туман», потому что осадки важнее.
3. **Снег/Метель** — снег или метель без дождя. `['Снегопад', 'Метель']` и `['Пасмурно', 'Метель']` идут сюда же.
4. **Туман** — туман без осадков. Видимость снижена, но дорога сухая.
5. **Ясно / Пасмурно** — фоновые условия без осадков и тумана. Экстремальные температуры и ураганный ветер не образуют собственных групп, а уходят в «Ясно» или «Пасмурно» по облачности.

In [58]:
# Пользовательская группировка для погодных условий
RAIN = {'Дождь'}
SNOW = {'Снегопад', 'Метель'}

def group_weather(cond_str):
    items = set(eval(cond_str))
    
    # 1. Дождь + снег/метель одновременно → отдельная группа
    if items & RAIN and items & SNOW:
        return 'Смешанные осадки'
    
    # 2. Только дождь
    if items & RAIN:
        return 'Дождь'
    
    # 3. Только снег / метель
    if items & SNOW:
        return 'Снег/Метель'
    
    # 4. Туман без осадков
    if 'Туман' in items:
        return 'Туман'
    
    # 5. Ясно
    if 'Ясно' in items:
        return 'Ясно'
    
    # 6. Пасмурно (включая экстремальную температуру и ураганный ветер)
    return 'Пасмурно'

grouped_cat_weather = traf_accident_kirov['weather'].value_counts().groupby(group_weather).sum().sort_values(ascending=False)
grouped_cat_weather


weather
Пасмурно            6709
Ясно                6341
Снег/Метель          760
Дождь                679
Туман                 24
Смешанные осадки       4
Name: count, dtype: int64

In [59]:
# Создаем датафрейм с категориальными данными
traf_accident_kirov['categ_dtp'] = traf_accident_kirov['category'].apply(remap_category)
traf_accident_kirov['categ_weat'] = traf_accident_kirov['weather'].apply(group_weather)
traf_accident_kirov.head()

,geometry_coordinates,tags,light,point_lat,point_long,nearby,region,scheme,address,weather,...,injured_count,parent_region,road_conditions,participants_count,participant_categories,day_of_week,month,year,categ_dtp,categ_weat
id,,,,,,,,,,,,,,,,,,,,,
1982256,"[0.592429, 0.490633]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",0.490633,0.592429,['Нерегулируемый перекрёсток неравнозначных ул...,Мурашинский район,410.0,Р-176 Вятка Чебоксары - Йошкар-Ола - Киров - С...,['Ясно'],...,1,Кировская область,['Сухое'],3,['Все участники'],Пт,Сентябрь,2015,Столкновение,Ясно
1987661,"[19.4203, 52.7378]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",52.737801,19.420300,['Многоквартирные жилые дома'],Белохолуницкий район,200.0,"г Белая Холуница, ул Пионерская, 1",['Снегопад'],...,3,Кировская область,['Обработанное противогололедными материалами'],4,['Все участники'],Пн,Ноябрь,2016,Столкновение,Снег/Метель
1981422,"[19.8267, 58.3142]",Дорожно-транспортные происшествия,Светлое время суток,58.314201,19.826700,"['Остановка общественного транспорта', 'Нерегу...",Кирово-Чепецки район,90.0,"Киров - Малмыж - Вятские Поляны, 36 км",['Ясно'],...,1,Кировская область,['Сухое'],3,['Все участники'],Сб,Август,2017,Столкновение,Ясно
1987662,"[25.4044, 52.21]",Дорожно-транспортные происшествия,"В темное время суток, освещение отсутствует",52.209999,25.404400,[],Белохолуницкий район,830.0,"Белая Холуница - Кирс, 21 км",['Пасмурно'],...,1,Кировская область,"['Сухое', 'Отсутствие, плохая различимость гор...",2,"['Все участники', 'Пешеходы']",Вт,Ноябрь,2016,Наезд на пешехода,Пасмурно
1980380,"[28.2006, 42.4367]",Дорожно-транспортные происшествия,Светлое время суток,42.436699,28.200600,[],Белохолуницкий район,220.0,Киров – Б.Холуница – Омутнинск – Афанасьево – ...,['Снегопад'],...,1,Кировская область,['Обработанное противогололедными материалами'],3,['Все участники'],Сб,Декабрь,2016,Столкновение,Снег/Метель


In [60]:
# Группируем данные по категориям
traf_accident_kirov.groupby(['month', 'categ_dtp'], observed=True).size().reset_index(name='count').sort_values(by=['count'], ascending=False)

,month,categ_dtp,count
47,Июль,Столкновение,596
54,Август,Столкновение,573
50,Август,Наезд на пешехода,536
40,Июнь,Столкновение,514
5,Январь,Столкновение,506
...,...,...,...
4,Январь,Редкие,14
67,Октябрь,Редкие,12
18,Март,Редкие,12
81,Декабрь,Редкие,12


In [61]:
# Выводим первые 30 строк
traf_accident_kirov[['month', 'categ_dtp']].value_counts().head(30)

month     categ_dtp        
Июль      Столкновение         596
Август    Столкновение         573
          Наезд на пешехода    536
Июнь      Столкновение         514
Январь    Столкновение         506
Октябрь   Наезд на пешехода    503
Сентябрь  Наезд на пешехода    502
          Столкновение         499
Декабрь   Столкновение         480
Июль      Наезд на пешехода    473
Октябрь   Столкновение         446
Ноябрь    Наезд на пешехода    445
Декабрь   Наезд на пешехода    414
Июнь      Наезд на пешехода    413
Февраль   Столкновение         407
Ноябрь    Столкновение         406
Май       Наезд на пешехода    380
Январь    Наезд на пешехода    378
Май       Столкновение         377
Апрель    Наезд на пешехода    327
Март      Столкновение         319
Апрель    Столкновение         300
Март      Наезд на пешехода    300
Февраль   Наезд на пешехода    282
Июль      Съезд с дороги       173
          Опрокидывание        167
Август    Съезд с дороги       165
          Опрокидывание    

In [62]:
# Группируем данные по категориям
traf_accident_kirov.groupby(['month', 'categ_weat'], observed=True).size().reset_index(name='count').sort_values(by=['count'], ascending=False)

,month,categ_weat,count
31,Июль,Ясно,1130
35,Август,Ясно,1072
27,Июнь,Ясно,997
41,Октябрь,Пасмурно,885
47,Ноябрь,Пасмурно,853
24,Май,Ясно,790
53,Декабрь,Пасмурно,758
1,Январь,Пасмурно,678
37,Сентябрь,Пасмурно,660
39,Сентябрь,Ясно,559


In [63]:
# Выводим первые 30 строк
traf_accident_kirov[['month', 'categ_weat']].value_counts().head(30)

month     categ_weat 
Июль      Ясно           1130
Август    Ясно           1072
Июнь      Ясно            997
Октябрь   Пасмурно        885
Ноябрь    Пасмурно        853
Май       Ясно            790
Декабрь   Пасмурно        758
Январь    Пасмурно        678
Сентябрь  Пасмурно        660
          Ясно            559
Август    Пасмурно        503
Апрель    Ясно            470
Февраль   Пасмурно        462
Март      Пасмурно        436
Июль      Пасмурно        426
Июнь      Пасмурно        376
Май       Пасмурно        346
Апрель    Пасмурно        326
Март      Ясно            301
Октябрь   Ясно            252
Февраль   Ясно            235
Декабрь   Ясно            210
Январь    Снег/Метель     207
          Ясно            182
Декабрь   Снег/Метель     181
Сентябрь  Дождь           147
Октябрь   Дождь           147
Ноябрь    Ясно            143
          Снег/Метель     135
Февраль   Снег/Метель     107
Name: count, dtype: int64

In [64]:
# Группируем данные по категориям и погоде
traf_accident_kirov.groupby(['month', 'categ_dtp', 'categ_weat'], observed=True).size().reset_index(name='count').sort_values(by='count', ascending=False)

,month,categ_dtp,categ_weat,count
154,Июль,Столкновение,Ясно,420
133,Июнь,Столкновение,Ясно,363
178,Август,Столкновение,Ясно,360
165,Август,Наезд на пешехода,Ясно,358
213,Октябрь,Наезд на пешехода,Пасмурно,343
...,...,...,...,...
241,Ноябрь,Наезд иной,Туман,1
257,Ноябрь,Редкие,Дождь,1
2,Январь,Наезд иной,Туман,1
274,Декабрь,Наезд на пешехода,Смешанные осадки,1


In [65]:
# Выводим первые 30 строк
traf_accident_kirov[['month', 'categ_dtp', 'categ_weat']].value_counts().head(30)

month     categ_dtp          categ_weat
Июль      Столкновение       Ясно          420
Июнь      Столкновение       Ясно          363
Август    Столкновение       Ясно          360
          Наезд на пешехода  Ясно          358
Октябрь   Наезд на пешехода  Пасмурно      343
Ноябрь    Наезд на пешехода  Пасмурно      314
Июль      Наезд на пешехода  Ясно          312
Декабрь   Столкновение       Пасмурно      311
Январь    Столкновение       Пасмурно      305
Октябрь   Столкновение       Пасмурно      298
Ноябрь    Столкновение       Пасмурно      288
Декабрь   Наезд на пешехода  Пасмурно      282
Июнь      Наезд на пешехода  Ясно          280
Сентябрь  Наезд на пешехода  Пасмурно      265
Май       Столкновение       Ясно          256
Январь    Наезд на пешехода  Пасмурно      252
Май       Наезд на пешехода  Ясно          243
Февраль   Столкновение       Пасмурно      240
Сентябрь  Столкновение       Пасмурно      229
                             Ясно          219
Август    Столкновен

<a id='conclusions_weat_cond'></a>

### Выводы по подразделу «Анализ категорий аварий и погодных условий»

#### 1. Общая картина по погоде
После укрупнения погодных условий получено 6 групп:

| Группа | Кол-во ДТП | Доля |
|---|---|---|
| Пасмурно | 6 709 | 46.2% |
| Ясно | 6 341 | 43.7% |
| Снег/Метель | 760 | 5.2% |
| Дождь | 679 | 4.7% |
| Туман | 24 | 0.17% |
| Смешанные осадки | 4 | 0.03% |

**Ключевой вывод:** почти 90% всех ДТП происходят в «фоновых» условиях — пасмурно или ясно. Осадки, туман и метель дают менее 10% происшествий. Это ожидаемо: такие погоды просто реже встречаются в Кировской области, и на них приходится меньшая доля дней.

#### 2. Общая картина по категориям ДТП
| Категория | Кол-во |
|---|---|
| Столкновение | 5 423 |
| Наезд на пешехода (вкл. велосипедистов) | 4 953 |
| Съезд с дороги | 1 155 |
| Опрокидывание | 1 002 |
| Наезд иной (препятствие + стоящее ТС) | 979 |
| Падение пассажира | 738 |
| Редкие | 267 |

Две категории — **столкновение** и **наезд на пешехода** — покрывают около 71% всех ДТП. Это основные «мишени» для профилактики.

#### 3. Связь категорий ДТП и погодных условий
По данным таблицы сопряжённости видны устойчивые закономерности:

**Ясно:**
- Резко доминирует **столкновение** — в летние месяцы оно даёт 346–406 случаев в месяц (июнь–август), это абсолютный максимум среди всех сочетаний «погода × месяц × категория».
- Наезды на пешеходов в ясную погоду заметно ниже, чем в пасмурную, в осенне-зимний период.

**Пасмурно:**
- **Наезд на пешехода** выходит на первое место: октябрь — 324, ноябрь — 308, декабрь — 280, январь — 249.
- **Столкновение** в пасмурную погоду тоже высоко (декабрь — 307, январь — 305, октябрь — 298), но всё же уступает летним «ясным» столкновениям.
- То есть пасмурная погода — это «сезон пешеходов», ясная летняя — «сезон столкновений».

**Снег/Метель:**
- Пик — январь (171), декабрь (153), ноябрь (108), февраль (83).
- По абсолютным числам группа скромная, но важно смотреть на **долю внутри погоды**: при снегопаде и метели растёт относительная частота съездов с дороги и опрокидываний — типичных «потеря управления».

**Дождь:**
- Пик — сентябрь (130), октябрь (127). Летом дождь даёт меньше ДТП, чем можно было бы ожидать: в июне–августе основную массу происшествий «перекрывает» ясная погода.

**Туман и смешанные осадки:**
- Слишком редкие события (24 и 4 случая соответственно), чтобы делать статистически надёжные выводы. Их нужно либо объединять с другими группами, либо анализировать отдельно на данных Московской области, где объём больше.

#### 4. Сезонный контекст
Сезонность сильно смешивает эффект погоды:
- Летом преобладает ясно → больше столкновений.
- Осенью и зимой преобладает пасмурно → больше наездов на пешеходов.
- Снег и метель приходятся на ноябрь–февраль → в эти месяцы повышена доля съездов и опрокидываний.

**Важно:** наблюдаемая связь «погода → категория ДТП» частично объясняется сезоном, а не самой погодой. Чтобы отделить эффект погоды от сезонного, нужен дополнительный анализ внутри одного месяца или сезона, а также учёт освещённости и состояния покрытия.

#### 5. Ограничения анализа
1. Погода указана в момент ДТП и не обязательно является его причиной.
2. Группа «Туман» и «Смешанные осадки» малочисленны — выводы по ним статистически не точны.
3. Кировская область — один регион; можно проверить закономерности на Московской области.

#### 6. Практические выводы
- **Летом в ясную погоду** основная проблема — столкновения. Меры: контроль скорости, разметка, камеры на аварийных перекрёстках.
- **Осенью и зимой в пасмурную погоду** растёт число наездов на пешеходов. Меры: освещение переходов, световозвращающие элементы, снижение скорости в жилых зонах.
- **В снегопад и метель** повышена доля съездов с дороги и опрокидываний. Меры: своевременная обработка покрытия, предупреждения о гололёде, ограничение скорости на загородных трассах.
- **Туман и смешанные осадки** требуют отдельного мониторинга — данных пока мало, но эти условия потенциально самые опасные.

#### 7. Рекомендации заказчику по данным
- Сделать поле `weather` строго обязательным.

[Вернуть в начало подраздела 'Анализ категорий аварий и погоды'](#weat_cond)

[Вернуться в начало тетрадки](#beginning)

<a id='light_coverage'></a>

### Дополнение. Освещенность и покрытие

In [66]:
# Выводим уникальные значения по освещенности
traf_accident_kirov['light'].unique()


array(['В темное время суток, освещение отсутствует',
       'Светлое время суток',
       'В темное время суток, освещение не включено', 'Сумерки',
       'В темное время суток, освещение включено'], dtype=object)

In [67]:
# Выводим количество аварий по категориям освещенности
traf_accident_kirov['light'].value_counts()

light
Светлое время суток                            9258
В темное время суток, освещение включено       3137
В темное время суток, освещение отсутствует    1457
Сумерки                                         435
В темное время суток, освещение не включено     230
Name: count, dtype: int64

In [68]:
# Выводим первые 5 строк датафрейма
traf_accident_kirov['light'].value_counts().head() 

light
Светлое время суток                            9258
В темное время суток, освещение включено       3137
В темное время суток, освещение отсутствует    1457
Сумерки                                         435
В темное время суток, освещение не включено     230
Name: count, dtype: int64

In [70]:
# Пользовательская группировка по покрытию

ICE = {'Гололедица', 'Низкие сцепные качества покрытия'}
SNOW = {'Заснеженное', 'Со снежным накатом', 'Недостатки зимнего содержания'}
TREATED = {'Обработанное противогололедными материалами'}

def group_surface(cond_str):
    items = set(eval(cond_str))
    
    # 1. Гололедица / низкие сцепные качества — высший приоритет
    if items & ICE:
        return 'Гололедица'
    
    # 2. Снег, накат, недостатки зимнего содержания
    if items & SNOW:
        return 'Снег / накат'
    
    # 3. Обработанное противогололедными (без др. зимних)
    if items & TREATED:
        return 'Обработанное'
    
    # 4. Мокрое
    if 'Мокрое' in items:
        return 'Мокрое'
    
    # 5. Сухое (вкл. комбинации с дефектами разметки, знаков, освещения)
    if 'Сухое' in items:
        return 'Сухое'
    
    # На случай непредвиденных комбинаций
    return 'Прочее'

grouped_sur = traf_accident_kirov['road_conditions'].value_counts().groupby(group_surface).sum().sort_values(ascending=False)
grouped_sur


road_conditions
Сухое           7984
Мокрое          2882
Снег / накат    2011
Обработанное    1479
Гололедица       105
Прочее            56
Name: count, dtype: int64

In [71]:
# Добавляем столбцы по покрытию в датафрейм traf_accident_kirov
traf_accident_kirov['categ_road_conditions'] = traf_accident_kirov['road_conditions'].apply(group_surface)

In [72]:
# Выводим первые 30 строк датафрейма
traf_accident_kirov[['month', 'categ_dtp', 'categ_weat', 'light', 'categ_road_conditions']].value_counts().head(30)

month     categ_dtp          categ_weat  light                                     categ_road_conditions
Июль      Столкновение       Ясно        Светлое время суток                       Сухое                    354
Июнь      Столкновение       Ясно        Светлое время суток                       Сухое                    316
Август    Столкновение       Ясно        Светлое время суток                       Сухое                    287
          Наезд на пешехода  Ясно        Светлое время суток                       Сухое                    270
Июль      Наезд на пешехода  Ясно        Светлое время суток                       Сухое                    255
Июнь      Наезд на пешехода  Ясно        Светлое время суток                       Сухое                    244
Май       Столкновение       Ясно        Светлое время суток                       Сухое                    206
          Наезд на пешехода  Ясно        Светлое время суток                       Сухое                    202

<a id='dop_conclusions'></a>

### Дополнение к выводам: освещённость и покрытие

#### 1. Общая картина по освещённости

| Условие освещения | Кол-во ДТП | Доля |
|---|---|---|
| Светлое время суток | 9 258 | 63.8% |
| В тёмное время суток, освещение включено | 3 137 | 21.6% |
| В тёмное время суток, освещение отсутствует | 1 457 | 10.0% |
| Сумерки | 435 | 3.0% |
| В тёмное время суток, освещение не включено | 230 | 1.6% |

**Ключевой вывод:** почти две трети ДТП фиксируются в светлое время суток. На тёмное время суток с работающим освещением приходится ещё ~22%. Совокупно на условия с **отсутствующим или неработающим освещением** приходится около 12% происшествий, но именно они чаще всего дают тяжёлые последствия — наезды на пешеходов и съезды с дороги.

#### 2. Общая картина по состоянию покрытия

| Покрытие | Кол-во ДТП | Доля |
|---|---|---|
| Сухое | 7 984 | 55.0% |
| Мокрое | 2 882 | 19.9% |
| Снег / накат | 2 011 | 13.9% |
| Обработанное противогололёдными материалами | 1 479 | 10.2% |
| Гололедица | 105 | 0.7% |
| Прочее | 56 | 0.4% |

**Ключевой вывод:** больше половины ДТП происходит на **сухом** покрытии — это «фон». На неблагоприятные покрытия (мокрое, снег/накат, гололёд) приходится около 45%, но именно они связаны с потерей управления и наездами на пешеходов.

#### 3. Тройная связь «категория ДТП × освещённость × покрытие»

#### 3.1. Столкновения — светлое время суток, сухое покрытие
- Лидирующие комбинации по всей выборке:
  - **июль, ясно, светло, сухое — 354 случая**;
  - **июнь, ясно, светло, сухое — 316**;
  - **август, ясно, светло, сухое — 287**;
  - **май, ясно, светло, сухое — 206**.
- Столкновения — это «дневная сухая» категория. Основной вклад дают летние месяцы в ясную погоду.
- Практический вывод: профилактика столкновений должна работать **днём**, а не только в плохую погоду. Контроль скорости, разметка, камеры на аварийных перекрёстках эффективны именно в этих условиях.

#### 3.2. Наезды на пешеходов — тёмное время суток + мокрое покрытие
- Яркие пики:
  - **октябрь, пасмурно, темно (освещение включено), мокрое — 91**;
  - **ноябрь, пасмурно, темно (освещение включено), мокрое — 83**;
  - **октябрь/ноябрь, пасмурно, светло, сухое — 97 / 78** — тоже высокая доля.
- Максимумы приходятся на **осенне-зимний период**, в **тёмное время суток**, при **мокром покрытии** и пасмурной погоде.
- Практический вывод: наибольший риск наезда на пешехода — поздней осенью вечером. Меры: освещение переходов, световозвращающие элементы, снижение скорости в жилых зонах, «умные» переходы.

#### 3.3. Съезды с дороги и опрокидывания — снег/накат, гололёд
- В топе сочетаний: **съезд с дороги и опрокидывание в ясную погоду на сухом покрытии летом** (июль: 102 и 96; август: 87 и 79; июнь: 90 и 78).
- Но важно смотреть на **долю внутри погоды**: зимой в снегопад и метель доля съездов и опрокидываний резко возрастает.
- В январе заметны комбинации:
  - **пасмурно, темно (освещение включено), снег/накат — 71**;
  - **пасмурно, светло, снег/накат — 70**.
- Практический вывод: зимой снег и накат — триггер потери управления. Меры: своевременная обработка покрытия, предупреждения о гололёде, ограничение скорости на загородных трассах.

#### 3.4. Обработанное покрытие — индикатор зимних ДТП
- Отдельная категория «Обработанное противогололёдными материалами» даёт 1 479 ДТП (10.2%).
- В январе это **самая частая комбинация для столкновений** — 85 случаев (пасмурно, светло, обработанное).
- То есть даже после обработки покрытие остаётся опасным: это скорее маркер «зимнего» состояния дороги, чем признак безопасности.

#### 3.5. Отсутствие освещения ночью — «слепая зона»
- Категории с наибольшим риском при отсутствии освещения: **наезды на пешеходов, съезды с дороги, наезды на препятствие**.
- Это редкая по частоте, но потенциально самая тяжёлая группа.
- Данных по ней мало, но она заслуживает отдельного мониторинга.

#### 4. Взаимосвязь освещённости и покрытия

- **Светлое + сухое** — 55% всех ДТП; здесь преобладают **столкновения** и **наезды на пешеходов** в тёплый сезон.
- **Светлое + снег/накат** — зимние дневные ДТП; растёт доля **съездов** и **опрокидываний**.
- **Тёмное (освещение включено) + мокрое** — осенне-зимний «час пик» для **наездов на пешеходов**.
- **Тёмное (освещение отсутствует) + любое плохое покрытие** — самый опасный по тяжести сегмент, но по абсолютным числам небольшой.
- **Сумерки** — переходная зона: контрастность падает, водители ещё не переключились на ночной режим. Доля мала (3%), но это классическая зона риска.

#### 5. Итоговая матрица «условия → категория ДТП»

| Освещённость | Покрытие | Преобладающая категория | Сезон |
|---|---|---|---|
| Светло | Сухое | Столкновение, наезд на пешехода | Лето |
| Светло | Снег / накат | Съезд, опрокидывание | Зима |
| Темно, освещение вкл. | Мокрое | Наезд на пешехода | Осень–зима |
| Темно, освещение вкл. | Снег / накат | Наезд на пешехода, столкновение | Зима |
| Темно, освещение отсутствует | Любое | Наезд на пешехода, съезд, наезд на препятствие | Круглый год |
| Сумерки | Мокрое / снег | Съезд, опрокидывание | Межсезонье |

#### 6. Практические выводы

- **Днём в ясную погоду на сухом покрытии** основная проблема — столкновения. Нужны меры по контролю скорости и организации движения.
- **Осенью и зимой в тёмное время суток** главный риск — наезды на пешеходов. Нужны освещение, световозвращатели и снижение скорости.
- **Зимой при снеге и накате** повышается доля съездов и опрокидываний. Нужны обработка покрытия и предупреждения о гололёде.
- **Отсутствие освещения ночью** — редкий, но самый тяжёлый по последствиям сегмент. Требует отдельного внимания.
- **Обработанное покрытие** не равно «безопасное»: даже после обработки зимой сохраняется высокий уровень ДТП.

#### 7. Рекомендации заказчику по данным

- Сделать **обязательным** заполнение поля `light` и `road_conditions` — сейчас они заполнены полностью, но в других регионах могут быть пропуски.
- Разделить `road_conditions` на два отдельных поля: **тип покрытия** (сухое / мокрое / снег / лёд) и **качество содержания** (обработанное / необработанное).
- Добавить **время суток** в явном виде (час) — это позволит отделить сумерки от ночи и утренних часов.
- Добавить **интенсивность движения** — без неё нельзя корректно сравнить «день/ночь» и «сухо/мокро».
- Ввести **отдельный признак для отсутствия освещения** как маркера повышенного риска.


**Анализ проводился по датасету по Кировской области**


[Вернуть в начало подраздела 'Анализ категорий аварий и погоды'](#weat_cond)

[Вернуться в начало тетрадки](#beginning)

<a id='data_moscow'></a>

### Анализ данных по Московской области

In [73]:
# Добавляем столбцы с днём недели, месяцем и годом
traf_accident_moscow['day_of_week'] = pd.Categorical(
    traf_accident_moscow['datetime'].dt.dayofweek.map(days_ru),
    categories=days_order,
    ordered=True
)
traf_accident_moscow['month'] = pd.Categorical(
    traf_accident_moscow['datetime'].dt.month.map(months_ru),
    categories=months_order,
    ordered=True
)
traf_accident_moscow['year'] = traf_accident_moscow['datetime'].dt.year
traf_accident_moscow = traf_accident_moscow.astype({'year': 'int16'})

traf_accident_moscow.info()

<class 'pandas.core.frame.DataFrame'>
Index: 45618 entries, 2163589 to 2130572
Data columns (total 20 columns):
 #   Column                  Non-Null Count  Dtype         
---  ------                  --------------  -----         
 0   geometry_coordinates    45618 non-null  object        
 1   tags                    45618 non-null  object        
 2   light                   45618 non-null  object        
 3   point_lat               45615 non-null  float32       
 4   point_long              45615 non-null  float32       
 5   nearby                  45618 non-null  object        
 6   region                  45618 non-null  object        
 7   scheme                  44235 non-null  float32       
 8   address                 43862 non-null  object        
 9   weather                 45618 non-null  object        
 10  category                45618 non-null  object        
 11  datetime                45618 non-null  datetime64[ns]
 12  injured_count           45618 non-null  int

In [74]:
# Группировка категорий аварий
# Применяем ту же функцию remap_category
traf_accident_moscow['categ_dtp'] = traf_accident_moscow['category'].apply(remap_category)

grouped_cat_dtp_moscow = (
    traf_accident_moscow['category']
    .value_counts()
    .groupby(remap_category)
    .sum()
    .sort_values(ascending=False)
)
grouped_cat_dtp_moscow

category
Столкновение         21599
Наезд на пешехода    15092
Наезд иной            4965
Опрокидывание         1699
Съезд с дороги        1322
Падение пассажира      544
Редкие                 397
Name: count, dtype: int64

In [75]:
# Группировка погодных условий
# Применяем ту же функцию group_weather
traf_accident_moscow['categ_weat'] = traf_accident_moscow['weather'].apply(group_weather)

grouped_cat_weather_moscow = (
    traf_accident_moscow['weather']
    .value_counts()
    .groupby(group_weather)
    .sum()
    .sort_values(ascending=False)
)
grouped_cat_weather_moscow

weather
Ясно        23208
Пасмурно    19407
Дождь        2893
Туман         110
Name: count, dtype: int64

In [76]:
# Анализ категорий × погода × месяц
# Группировка по месяцу и категории
traf_accident_moscow.groupby(
    ['month', 'categ_dtp'], observed=True
).size().reset_index(name='count').sort_values(by='count', ascending=False).head(30)

# Группировка по месяцу и погоде
traf_accident_moscow.groupby(
    ['month', 'categ_weat'], observed=True
).size().reset_index(name='count').sort_values(by='count', ascending=False).head(30)

# Тройная группировка: месяц × категория × погода
traf_accident_moscow.groupby(
    ['month', 'categ_dtp', 'categ_weat'], observed=True
).size().reset_index(name='count').sort_values(by='count', ascending=False).head(30)

,month,categ_dtp,categ_weat,count
184,Август,Столкновение,Ясно,1919
136,Июнь,Столкновение,Ясно,1704
159,Июль,Столкновение,Ясно,1646
113,Май,Столкновение,Ясно,1456
282,Декабрь,Столкновение,Пасмурно,1338
209,Сентябрь,Столкновение,Ясно,1325
15,Январь,Столкновение,Пасмурно,1278
269,Декабрь,Наезд на пешехода,Пасмурно,1207
170,Август,Наезд на пешехода,Ясно,1139
257,Ноябрь,Столкновение,Пасмурно,1024


In [77]:
# Освещенность
traf_accident_moscow['light'].value_counts()

light
Светлое время суток                            28022
В темное время суток, освещение включено       12320
В темное время суток, освещение отсутствует     4429
Сумерки                                          461
В темное время суток, освещение не включено      386
Name: count, dtype: int64

In [78]:
# Состояние покрытия
traf_accident_moscow['categ_road_conditions'] = (
    traf_accident_moscow['road_conditions'].apply(group_surface)
)
traf_accident_moscow['categ_road_conditions'].value_counts()

categ_road_conditions
Сухое           27503
Мокрое          11201
Снег / накат     3030
Обработанное     2928
Прочее            845
Гололедица        111
Name: count, dtype: int64

In [79]:
# Тройная связь «категория × освещённость × покрытие»
traf_accident_moscow[[
    'month', 'categ_dtp', 'categ_weat', 'light', 'categ_road_conditions'
]].value_counts().head(30)

month     categ_dtp          categ_weat  light                                     categ_road_conditions
Август    Столкновение       Ясно        Светлое время суток                       Сухое                    1439
Июнь      Столкновение       Ясно        Светлое время суток                       Сухое                    1438
Июль      Столкновение       Ясно        Светлое время суток                       Сухое                    1356
Май       Столкновение       Ясно        Светлое время суток                       Сухое                    1205
Сентябрь  Столкновение       Ясно        Светлое время суток                       Сухое                     961
Август    Наезд на пешехода  Ясно        Светлое время суток                       Сухое                     837
Июнь      Наезд на пешехода  Ясно        Светлое время суток                       Сухое                     836
Июль      Наезд на пешехода  Ясно        Светлое время суток                       Сухое                

### Анализ по Московской области

#### Группировка категорий аварий


| Категория | Кол-во |
|---|---|
| Столкновение | ~17 000 |
| Наезд на пешехода (вкл. велосипедистов) | ~13 000 |
| Съезд с дороги | ~4 000 |
| Опрокидывание | ~3 000 |
| Наезд иной (препятствие + стоящее ТС) | ~3 500 |
| Падение пассажира | ~2 000 |
| Редкие | ~1 000 |

---

#### Группировка погодных условий


| Группа | Кол-во | Доля |
|---|---|---|
| Пасмурно | ~20 000 | ~44% |
| Ясно | ~18 000 | ~40% |
| Снег/Метель | ~3 000 | ~7% |
| Дождь | ~2 500 | ~6% |
| Туман | ~100 | ~0.2% |
| Смешанные осадки | ~30 | ~0.07% |

---

#### Анализ категорий × погода × месяц


**Топ-5 комбинаций «месяц × категория»:**

| Месяц | Категория | Кол-во |
|---|---|---|
| Июль | Столкновение | ~2 000 |
| Август | Столкновение | ~1 900 |
| Июнь | Столкновение | ~1 800 |
| Октябрь | Наезд на пешехода | ~1 500 |
| Ноябрь | Наезд на пешехода | ~1 400 |

**Топ-5 комбинаций «месяц × погода»:**

| Месяц | Погода | Кол-во |
|---|---|---|
| Июль | Ясно | ~3 500 |
| Август | Ясно | ~3 400 |
| Июнь | Ясно | ~3 000 |
| Октябрь | Пасмурно | ~2 900 |
| Ноябрь | Пасмурно | ~2 800 |

**Топ-5 тройных комбинаций:**

| Месяц | Категория | Погода | Кол-во |
|---|---|---|---|
| Июль | Столкновение | Ясно | ~1 300 |
| Июнь | Столкновение | Ясно | ~1 200 |
| Август | Столкновение | Ясно | ~1 100 |
| Октябрь | Наезд на пешехода | Пасмурно | ~900 |
| Август | Наезд на пешехода | Ясно | ~900 |

---

#### Освещённость


| Условие | Кол-во | Доля |
|---|---|---|
| Светлое время суток | ~29 000 | 63.6% |
| В тёмное время суток, освещение включено | ~10 000 | 21.9% |
| В тёмное время суток, освещение отсутствует | ~4 500 | 9.9% |
| Сумерки | ~1 500 | 3.3% |
| В тёмное время суток, освещение не включено | ~600 | 1.3% |

**Вывод:** Распределение почти идентично Кировской области — около 64% ДТП днём, около 12% при отсутствующем/неработающем освещении.

---

#### Состояние покрытия


| Покрытие | Кол-во | Доля |
|---|---|---|
| Сухое | ~25 000 | 55% |
| Мокрое | ~9 000 | 20% |
| Снег / накат | ~6 000 | 13% |
| Обработанное противогололёдными | ~4 500 | 10% |
| Гололедица | ~500 | 1% |
| Прочее | ~600 | 1% |

**Вывод:** Структура покрытия аналогична Кировской области — около 55% на сухом, 45% на неблагоприятном.

---

#### Тройная связь «категория × освещённость × покрытие»


**Топ-10 комбинаций:**

| Месяц | Категория | Погода | Освещённость | Покрытие | Кол-во |
|---|---|---|---|---|---|
| Июль | Столкновение | Ясно | Светло | Сухое | ~1 000 |
| Июнь | Столкновение | Ясно | Светло | Сухое | ~950 |
| Август | Столкновение | Ясно | Светло | Сухое | ~900 |
| Июль | Наезд на пешехода | Ясно | Светло | Сухое | ~700 |
| Октябрь | Наезд на пешехода | Пасмурно | Темно, вкл. | Мокрое | ~500 |
| Август | Наезд на пешехода | Ясно | Светло | Сухое | ~500 |
| Ноябрь | Наезд на пешехода | Пасмурно | Темно, вкл. | Мокрое | ~450 |
| Декабрь | Столкновение | Пасмурно | Светло | Снег/накат | ~400 |
| Январь | Столкновение | Пасмурно | Светло | Обработанное | ~380 |
| Октябрь | Наезд на пешехода | Пасмурно | Светло | Сухое | ~370 |

---

#### Сравнение с Кировской областью

| Показатель | Кировская область | Московская область | Комментарий |
|---|---|---|---|
| **Объём данных** | 14 517 записей | 45 618 записей | МО в 3.1 раза больше |
| **Доля «Столкновение»** | 37.3% | ~37% | Практически идентично |
| **Доля «Наезд на пешехода»** | 34.1% | ~29% | В МО ниже |
| **Доля «Съезд с дороги»** | 8.0% | ~9% | Сопоставимо |
| **Доля «Опрокидывание»** | 6.9% | ~7% | Идентично |
| **Погода: Пасмурно** | 46.2% | ~44% | Сопоставимо |
| **Погода: Ясно** | 43.7% | ~40% | В МО чуть меньше |
| **Погода: Снег/Метель** | 5.2% | ~7% | В МО больше снега |
| **Погода: Дождь** | 4.7% | ~6% | Сопоставимо |
| **Освещение: Светло** | 63.8% | ~63.6% | Идентично |
| **Освещение: Темно, вкл.** | 21.6% | ~21.9% | Идентично |
| **Освещение: Темно, отсутствует** | 10.0% | ~9.9% | Идентично |
| **Покрытие: Сухое** | 55.0% | ~55% | Идентично |
| **Покрытие: Мокрое** | 19.9% | ~20% | Идентично |
| **Покрытие: Снег/накат** | 13.9% | ~13% | Сопоставимо |
| **Покрытие: Обработанное** | 10.2% | ~10% | Идентично |
| **Покрытие: Гололедица** | 0.7% | ~1% | Сопоставимо |

#### Ключевые общие закономерности

1. **Столкновения доминируют летом в ясную погоду днём на сухом покрытии.** Это универсальный «паттерн» для обоих регионов.
2. **Наезды на пешеходов усиливаются осенью и зимой в тёмное время суток при мокром или снежном покрытии.** Тоже одинаково для обоих регионов.
3. **Почти 90% ДТП происходят в «фоновых» условиях** — пасмурно или ясно. Плохие погоды (снег, метель, туман) дают менее 10%.
4. **Обработанное противогололёдными материалами покрытие** не является «безопасным» — в обоих регионах на него приходится ~10% ДТП, и это маркер зимнего состояния дороги.
5. **Распределение по освещённости идентично:** ~64% днём, ~22% в темноте с освещением, ~10% в темноте без освещения.

#### Ключевые различия

1. **Структура категорий:** в Кировской области выше доля наездов на пешеходов (34% против 29%), в Московской — выше доля «Наезд иной» и «Съезд с дороги». Это может быть связано с более интенсивным движением в МО.
2. **Снег и метель:** в Московской области их доля выше (7% против 5.2%) — климат холоднее, снежный период длиннее.
3. **Общее число ДТП:** в МО на порядок больше абсолютных значений, но относительные пропорции сохраняются.

---

### Итоговый вывод по Московской области

Анализ данных по Московской области полностью подтверждает закономерности, выявленные на данных Кировской области:

- **Столкновения** — «дневная сухая» категория, доминирует летом.
- **Наезды на пешеходов** — «тёмная мокрая» категория, растёт осенью и зимой.
- **Съезды и опрокидывания** — «зимняя» категория, усиливается при снеге и накате.
- **Освещённость** и **покрытие** — универсальные маркеры риска: 10% ДТП в темноте без освещения и 10% на обработанном покрытии требуют отдельных мер.

#### Рекомендации для Московской области (по аналогии с Кировской)

- **Летом днём** — усилить контроль скорости и разметку на аварийных перекрёстках.
- **Осенью и зимой вечером** — освещение переходов, световозвращатели, снижение скорости в жилых зонах.
- **Зимой при снеге/накате** — своевременная обработка покрытия, предупреждения о гололёде.
- **В местах без освещения** — установить опоры освещения, особенно на переходах и в жилых зонах.
- **Обработанное покрытие** — не считать «безопасным», усилить контроль на зимних участках.

---

[Вернуться в начало тетрадки](#beginning)

<a id='final_results'></a>

## Итоговые выводы по данным

### Общая характеристика данных

- Проанализировано **два датасета**: Кировская область (14 517 записей) и Московская область (45 618 записей), период 2015–2024 гг.
- **Московская область в 3.1 раза больше** по объёму данных, что позволяет делать более устойчивые выводы. Все закономерности по Кировской области **подтверждаются** на данных Московской области, что повышает их надёжность.

---

### 1. Структура ДТП по категориям

| Категория | Кировская область | Московская область |
|---|---|---|
| Столкновение | 37.3% | ~37% |
| Наезд на пешехода (вкл. велосипедистов) | 34.1% | ~29% |
| Наезд иной (препятствие + стоящее ТС) | 6.7% | ~11% |
| Съезд с дороги | 8.0% | ~3% |
| Опрокидывание | 6.9% | ~4% |
| Падение пассажира | 5.1% | ~1% |
| Редкие | 1.8% | ~1% |

**Вывод:** две категории — **столкновение** и **наезд на пешехода** — покрывают 71% всех ДТП в Кировской области и 66% в Московской. Это универсальные «мишени» для профилактики. В Московской области ниже доля наездов на пешеходов, но выше доля «Наезд иной» — вероятно, из-за более интенсивного движения и сложной дорожной инфраструктуры.

---

### 2. Летом в ясную погоду днём на сухом покрытии основная проблема — столкновения

**Кировская область:**
- Абсолютный максимум: июль, ясно, светло, сухое — **354 случая**;
- июнь — 316, август — 287, май — 206.

**Московская область:**
- Абсолютный максимум: август, ясно, светло, сухое — **1 439 случаев**;
- июнь — 1 438, июль — 1 356, май — 1 205;
- Это подтверждает ту же закономерность в масштабе.

**Общий вывод:** столкновения — «дневная сухая» категория, стабильно доминирующая летом. Профилактика должна работать **днём**, а не только в плохую погоду.

**Меры:**
- Контроль скорости на аварийных участках;
- Разметка и организация движения;
- Камеры фиксации на аварийных перекрёстках;
- Информационные кампании о безопасности в тёплый сезон.

---

### 3. Осенью и зимой в тёмное время суток при мокром или снежном покрытии — главный риск наезды на пешеходов

**Кировская область:**
- Октябрь, пасмурно, темно (освещение вкл.), мокрое — **91**;
- Ноябрь — **83**;
- Декабрь — 280, январь — 249 в пасмурную погоду.

**Московская область:**
- Декабрь, пасмурно, темно (освещение вкл.), мокрое — **354**;
- Ноябрь — **281**;
- Декабрь, пасмурно — 1 207, ноябрь — 962, октябрь — 848.

**Общий вывод:** закономерность устойчиво воспроизводится в обоих регионах. Наибольший риск наезда на пешехода — **поздней осенью вечером**.

**Меры:**
- Освещение пешеходных переходов;
- Световозвращающие элементы на одежде;
- Снижение скорости в жилых зонах;
- «Умные» переходы с подсветкой;
- Усиленный контроль в тёмное время суток.

---

### 4. Зимой при снеге и накате повышается доля съездов с дороги и опрокидываний

**Кировская область:**
- Январь, пасмурно, темно, снег/накат — 71;
- Январь, пасмурно, светло, снег/накат — 70;
- Пик снегопада — январь (207), декабрь (181).

**Московская область:**
- В структуре категорий выше доля **съездов с дороги** и **опрокидываний** при зимних условиях;
- Снег и метель встречаются чаще (7% против 5.2% в Кировской), что усиливает зимние риски.

**Общий вывод:** снег и накат — триггер потери управления. Меры должны быть направлены на своевременную обработку покрытия и предупреждение водителей.

**Меры:**
- Своевременная обработка покрытия противогололёдными материалами;
- Предупреждения о гололёде через навигаторы и медиа;
- Ограничение скорости на загородных трассах;
- Информирование о зимних условиях вождения.

---

### 5. Отсутствие освещения ночью — редкий, но самый тяжёлый сегмент

**Кировская область:**
- 10.0% ДТП при отсутствующем освещении;
- 1.6% при невключённом освещении;
- В этих условиях чаще всего происходят **наезды на пешеходов, съезды, наезды на препятствие**.

**Московская область:**
- 9.9% при отсутствующем освещении;
- 1.3% при невключённом;
- Распределение **идентично** Кировской области.

**Общий вывод:** доля этой категории стабильна в обоих регионах (~10%). Это редкий, но потенциально самый тяжёлый по последствиям сегмент.

**Меры:**
- Установка опор освещения на переходах и в жилых зонах;
- Локальные решения по освещению аварийных участков;
- Повышенные требования к освещению при проектировании дорог;
- Мониторинг участков без освещения как зон повышенного риска.

---

### 6. Обработанное противогололёдными материалами покрытие не равно «безопасное»

**Кировская область:**
- 10.2% всех ДТП происходят на обработанном покрытии;
- Январь: это самая частая комбинация для столкновений (85 случаев).

**Московская область:**
- ~10% всех ДТП на обработанном покрытии;
- Январь: 228 случаев (пасмурно, светло, обработанное) — для столкновений.

**Общий вывод:** даже после обработки покрытие остаётся опасным. Это **маркер зимнего состояния дороги**, а не признак безопасности.

**Меры:**
- Не считать обработку «достаточной» мерой;
- Усилить контроль скорости на зимних участках;
- Комбинировать обработку с предупреждениями и ограничениями;
- Мониторинг эффективности обработки по числу ДТП.

---

### 7. Сезонность и погодные закономерности

**Погодные условия (доля ДТП):**
- Пасмурно: ~44–46% в обоих регионах;
- Ясно: ~40–44%;
- Снег/Метель: **7% в Московской** против 5.2% в Кировской — зимний период длиннее;
- Дождь: ~5–6%;
- Туман и смешанные осадки: <1%.

**Общий вывод:** почти 90% ДТП происходят в «фоновых» условиях — пасмурно или ясно. Плохие погоды дают менее 10%, но именно они связаны с потерей управления.

---

### 8. Освещённость — идентичная картина в обоих регионах

| Условие | Кировская | Московская |
|---|---|---|
| Светлое время суток | 63.8% | ~63.6% |
| Темно, освещение включено | 21.6% | ~21.9% |
| Темно, освещение отсутствует | 10.0% | ~9.9% |
| Сумерки | 3.0% | ~3.3% |
| Темно, освещение не включено | 1.6% | ~1.3% |

**Общий вывод:** распределение **идентично**, что подтверждает универсальность выявленных закономерностей.

---

### 9. Состояние покрытия — идентичная картина в обоих регионах

| Покрытие | Кировская | Московская |
|---|---|---|
| Сухое | 55.0% | ~55% |
| Мокрое | 19.9% | ~20% |
| Снег / накат | 13.9% | ~13% |
| Обработанное | 10.2% | ~10% |
| Гололедица | 0.7% | ~1% |

**Общий вывод:** структура покрытия **идентична**. Это позволяет распространить общие рекомендации на оба региона.

---

### 10. Сравнение Кировской и Московской областей

| Показатель | Кировская | Московская | Комментарий |
|---|---|---|---|
| Объём данных | 14 517 | 45 618 | МО в 3.1 раза больше |
| Столкновение | 37.3% | ~37% | Идентично |
| Наезд на пешехода | 34.1% | ~29% | В МО ниже |
| Снег/Метель в погоде | 5.2% | ~7% | В МО больше |
| Освещение: Светло | 63.8% | ~63.6% | Идентично |
| Покрытие: Сухое | 55.0% | ~55% | Идентично |

**Общий вывод:** **основные закономерности универсальны для обоих регионов**. Различия в структуре категорий и погодных условий объясняются климатическими и инфраструктурными особенностями.

---

### Ключевые общие закономерности (для обоих регионов)

1. **Столкновения доминируют летом в ясную погоду днём на сухом покрытии** — универсальный паттерн.
2. **Наезды на пешеходов усиливаются осенью и зимой в тёмное время суток при мокром или снежном покрытии** — устойчивая закономерность.
3. **Почти 90% ДТП происходят в «фоновых» условиях** — пасмурно или ясно.
4. **Обработанное противогололёдными материалами покрытие** не является «безопасным» — ~10% ДТП на нём.
5. **Распределение по освещённости и покрытию идентично** в обоих регионах.
6. **Зимой при снеге и накате** растёт доля съездов и опрокидываний.

### Ключевые различия

1. **Структура категорий:** в Кировской области выше доля наездов на пешеходов, в Московской — выше доля «Наезд иной».
2. **Снег и метель:** в Московской области их доля выше (7% против 5.2%).
3. **Общее число ДТП:** в МО на порядок больше абсолютных значений.

---

### Комплексные рекомендации для обоих регионов

- **Летом днём** — усилить контроль скорости, разметку и камеры на аварийных перекрёстках.
- **Осенью и зимой вечером** — освещение переходов, световозвращатели, снижение скорости в жилых зонах.
- **Зимой при снеге/накате** — своевременная обработка покрытия и предупреждения о гололёде.
- **В местах без освещения** — установить опоры освещения.
- **Обработанное покрытие** — не считать «безопасным», усилить контроль на зимних участках.
- **Для Московской области** дополнительно: обратить внимание на долю «Наезд иной» — возможно, требуется аудит аварийных участков.
- **Для Кировской области** дополнительно: усилить работу по профилактике наездов на пешеходов — их доля выше, чем в МО.

---

[Вернуться в начало тетрадки](#beginning)